<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>ACMG-AMP classification for germline variant pathogenicity | Cancer Data Diaries</title>
<meta name="keywords" content="ACMG, germline, variant interpretation">
<meta name="description" content="Germline variant effect interpretation is challenging Assessing the clinical significance of germline variants is an important task in genetics. Identification of pathogenic germline variants may affect the diagnosis and treatment decisions and hence potentially improve and personalize the patient care. In addition, the identification of such heritable risk is important for the family of the patients as they now may be provided genetic counseling and cancer surveillance if they are found to be carriers of the same variant.">
<meta name="author" content="Me">
<link rel="canonical" href="http://localhost:1313/posts/germline-variant-classification/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.b648782d138748285f2b1210a71b68efe4fa23d8cf11526df0b2fb74770c0de8.css" integrity="sha256-tkh4LROHSChfKxIQpxto7&#43;T6I9jPEVJt8LL7dHcMDeg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" hreflang="en" href="http://localhost:1313/posts/germline-variant-classification/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
  

<meta property="og:title" content="ACMG-AMP classification for germline variant pathogenicity" />
<meta property="og:description" content="Germline variant effect interpretation is challenging Assessing the clinical significance of germline variants is an important task in genetics. Identification of pathogenic germline variants may affect the diagnosis and treatment decisions and hence potentially improve and personalize the patient care. In addition, the identification of such heritable risk is important for the family of the patients as they now may be provided genetic counseling and cancer surveillance if they are found to be carriers of the same variant." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/posts/germline-variant-classification/" />
<meta property="og:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta property="article:section" content="posts" />
<meta property="article:published_time" content="2024-08-03T15:29:55+09:00" />
<meta property="article:modified_time" content="2024-08-03T15:29:55+09:00" /><meta property="og:site_name" content="Cancer Data Diaries" />


<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta name="twitter:title" content="ACMG-AMP classification for germline variant pathogenicity"/>
<meta name="twitter:description" content="Germline variant effect interpretation is challenging Assessing the clinical significance of germline variants is an important task in genetics. Identification of pathogenic germline variants may affect the diagnosis and treatment decisions and hence potentially improve and personalize the patient care. In addition, the identification of such heritable risk is important for the family of the patients as they now may be provided genetic counseling and cancer surveillance if they are found to be carriers of the same variant."/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position":  1 ,
      "name": "Posts",
      "item": "http://localhost:1313/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "ACMG-AMP classification for germline variant pathogenicity",
      "item": "http://localhost:1313/posts/germline-variant-classification/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "ACMG-AMP classification for germline variant pathogenicity",
  "name": "ACMG-AMP classification for germline variant pathogenicity",
  "description": "Germline variant effect interpretation is challenging Assessing the clinical significance of germline variants is an important task in genetics. Identification of pathogenic germline variants may affect the diagnosis and treatment decisions and hence potentially improve and personalize the patient care. In addition, the identification of such heritable risk is important for the family of the patients as they now may be provided genetic counseling and cancer surveillance if they are found to be carriers of the same variant.",
  "keywords": [
    "ACMG", "germline", "variant interpretation"
  ],
  "articleBody": "Germline variant effect interpretation is challenging Assessing the clinical significance of germline variants is an important task in genetics. Identification of pathogenic germline variants may affect the diagnosis and treatment decisions and hence potentially improve and personalize the patient care. In addition, the identification of such heritable risk is important for the family of the patients as they now may be provided genetic counseling and cancer surveillance if they are found to be carriers of the same variant.\nDetermining variant pathogenicity can be challenging due to the complexity and diversity of genetic variations. Evidence for variant interpretation may be scarce or conflicting which often leads variants to be classified as variants of uncertain significance (VUS).\nHow to determine the clinical significance, pathogenicity, of a variant?\nEvidence-based system for variant interpretation The American College of Medical Genetics and Genomics and the Association for Molecular Pathology (ACMG-AMP) (Richards et al., 2015) has developed guidelines for variant interpretation and is currently widely used system in this task.\nThe guidelines provide a standardized and systematic framework for evaluating the pathogenicity of germline variants. It is based on a system that looks through evidence for the variant and scores the evidence based on how strongly it supports pathogenicity or benignity.\nFor final variant interpretation the evidence for the variant in question is combined and variant will be assigned to one of the five pathogenicity classes: Pathogenic (P), Likely pathogenic (LP), Variant of uncertain significance (VUS), Likely benign (LB) or Benign (B) (Figure 1).\nFigure 1. ACMG classification system Evidence used for variant interpretation includes:\npopulation control data (eg. gnomAD (Karczewski, et al., 2020)) in silico /computation predictions (such as pathogenicity prediction score AlphaMissense (Cheng et al., 2023)) Functional data (eg. Hotspot regions) Clinical data/earlier findings (eg. ClinVar (Landrum et al., 2014)) In addition, the variant may be evaluated based on segregation, de novo, and allelic data if such information is available.\nAll the ACMG-AMP guideline evidence and their perceived strength can be seen from table 1.\nTable 1. Richards et al., ACMG classification rules Implementing the ACMG-AMP guidelines Multiple tools have been created to implement these guidelines, such as TAPES (Xavier et al., 2019). Recently, tools utilizing machine learning methods for variant prediction, such as MLVar (Nicora et al., 2022), have been published.These tools have been created to help automate the variant interpretation as going through a large number of variants manually is time-consuming task. Automation creates some restrictions because it is not possible to automate evidence such as segregation data which is patient specific. Some of these problems can be dealt with by making the user provide the additional data separately for the analysis.\nChallenges of the automated classification system Automation can speed up the analysis and help to filter and decrease the variant set to those predicted as pathogenic. However, because the rules allow for some subjectivity in the interpretation, there are inconsistencies between clinical laboratories (Kim et al., 2019). In addition, the varying prevalence of variants in different populations poses a challenge when determining whether a variant is rare or not.\nAnother huge challenge in the guidelines is the rules using lists of genes with some ‘known’ qualities. For example, very strong rule PVS1 is applied for a null variant located in a gene for which loss-of-function (LOF) is a known mechanism of disease. In practice, many tools interpret this rule by constructing a list of genes for which there are certain number of earlier identified pathogenic null variants. Benign rule, BP1 missense variant in a gene where only truncating variants cause disease, may reduce the importance of new pathogenic missense variant findings because in many cases disease associated genes have been found by identifying variants with clear, big impact (truncating/null variants).\nIt’s important to be aware of the potential caveats and limitations of the classification system, especially when identifying pathogenic variants in a clinical setting. ACMG system should work as a guideline to restrict the number of variants of interest for further inspection and not completely erase the manual work required to assess the importance of variants on patient level.\nMore work should still be done to clarify and develop the guidelines in order to reduce the inconsistencies. This is important also for the machine learning based methods because many tools use the variant databases, consisting of variants with manual or automated ACMG classifications, as training data.\nReferences Richards S, Aziz N, Bale S, et al. Standards and guidelines for the interpretation of sequence variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology. Genet Med. 17(5):405-424. (2015) doi:10.1038/gim.2015.30\nKarczewski KJ., Francioli LC., Tiao, G. et al. The mutational constraint spectrum quantified from variation in 141,456 humans. Nature ****581, 434–443 (2020). https://doi.org/10.1038/s41586-020-2308-7\nCheng J et al. Accurate proteome-wide missense variant effect prediction with AlphaMissense. Science 381,eadg7492(2023). DOI:10.1126/science.adg7492\nLandrum MJ, Lee JM, Riley GR, Jang W, Rubinstein WS, Church DM, Maglott DR. ClinVar: public archive of relationships among sequence variation and human phenotype. Nucleic Acids Res. 42(1):D980-5 (2014). doi: 10.1093/nar/gkt1113\nXavier A, Scott RJ, Talseth-Palmer BA. TAPES: A tool for assessment and prioritisation in exome studies. PLoS Comput Biol 15(10):e1007453 (2019). doi: 10.1371/journal.pcbi.1007453\nNicora, G., Zucca, S., Limongelli, I. et al. A machine learning approach based on ACMG/AMP guidelines for genomic variant classification and prioritization. Sci Rep 12, 2517 (2022). https://doi.org/10.1038/s41598-022-06547-3\nKim Y, Ki C, Jang M. Challenges and Considerations in Sequence Variant Interpretation for Mendelian Disorders. Ann Lab Med 39:421-429 (2019). https://doi.org/10.3343/alm.2019.39.5.421\n",
  "wordCount" : "911",
  "inLanguage": "en",
  "image": "http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E","datePublished": "2024-08-03T15:29:55+09:00",
  "dateModified": "2024-08-03T15:29:55+09:00",
  "author":{
    "@type": "Person",
    "name": "Me"
  },
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://localhost:1313/posts/germline-variant-classification/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Cancer Data Diaries",
    "logo": {
      "@type": "ImageObject",
      "url": "http://localhost:1313/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Home (Alt + H)">
                <img src="http://localhost:1313/icon.png" alt="" aria-label="logo"
                    height="35">Home</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/categories/" title="Categories">
                    <span>Categories</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search/" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/tags/" title="Tags">
                    <span>Tags</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main">
<main>
    <article>
    <header>
        <h1>ACMG-AMP classification for germline variant pathogenicity</h1>
    </header>

        


        <h2 id="germline-variant-effect-interpretation-is-challenging">Germline variant effect interpretation is challenging</h2>
<p>Assessing the clinical significance of germline variants is an important task in genetics. Identification of pathogenic germline variants may affect the diagnosis and treatment decisions and hence potentially improve and personalize the patient care. In addition, the identification of such heritable risk is important for the family of the patients as they now may be provided genetic counseling and cancer surveillance if they are found to be carriers of the same variant.</p>
<p>Determining variant pathogenicity can be challenging due to the complexity and diversity of genetic variations. Evidence for variant interpretation may be scarce or conflicting which often leads variants to be classified as variants of uncertain significance (VUS).</p>
<p>How to determine the clinical significance, pathogenicity, of a variant?</p>
<h2 id="evidence-based-system-for-variant-interpretation">Evidence-based system for variant interpretation</h2>
<p>The American College of Medical Genetics and Genomics and the Association for Molecular Pathology (ACMG-AMP) (Richards et al., 2015) has developed guidelines for variant interpretation and is currently widely used system in this task.</p>
<p>The guidelines provide a standardized and systematic framework for evaluating the pathogenicity of germline variants. It is based on a system that looks through evidence for the variant and scores the evidence based on how strongly it supports pathogenicity or benignity.</p>
<p>For final variant interpretation the evidence for the variant in question is combined and variant will be assigned to one of the five pathogenicity classes: Pathogenic (P), Likely pathogenic (LP), Variant of uncertain significance (VUS), Likely benign (LB) or Benign (B) (Figure 1).</p>
<figure>
    <img loading="lazy" src="images/ACMGpipeline.jpg"/> <figcaption>
            Figure 1. ACMG classification system
        </figcaption>
</figure>

<p>Evidence used for variant interpretation includes:</p>
<ul>
<li>population control data (eg. gnomAD (Karczewski, et al., 2020))</li>
<li><em>in silico</em> /computation predictions (such as pathogenicity prediction score AlphaMissense (Cheng et al., 2023))</li>
<li>Functional data (eg. Hotspot regions)</li>
<li>Clinical data/earlier findings (eg. ClinVar (Landrum et al., 2014))</li>
</ul>
<p>In addition, the variant may be evaluated based on segregation, <em>de novo</em>, and allelic data if such information is available.</p>
<p>All the ACMG-AMP guideline evidence and their perceived strength can be seen from table 1.</p>
<figure>
    <img loading="lazy" src="acmgrules.jpg"/> <figcaption>
            Table 1. Richards et al., ACMG classification rules
        </figcaption>
</figure>

<h2 id="implementing-the-acmg-amp-guidelines">Implementing the ACMG-AMP guidelines</h2>
<p>Multiple tools have been created to implement these guidelines, such as TAPES (Xavier et al., 2019). Recently, tools utilizing machine learning methods for variant prediction, such as MLVar (Nicora et al., 2022), have been published.These tools have been created to help automate the variant interpretation as going through a large number of variants manually is time-consuming task. Automation creates some restrictions because it is not possible to automate evidence such as segregation data which is patient specific. Some of these problems can be dealt with by making the user provide the additional data separately for the analysis.</p>
<h2 id="challenges-of-the-automated-classification-system">Challenges of the automated classification system</h2>
<p>Automation can speed up the analysis and help to filter and decrease the variant set to those predicted as pathogenic. However, because the rules allow for some subjectivity in the interpretation, there are inconsistencies  between clinical laboratories (Kim et al., 2019). In addition, the varying prevalence of variants in different populations poses a challenge when determining whether a variant is rare or not.</p>
<p>Another huge challenge in the guidelines is the rules using lists of genes with some ‘known’ qualities. For example, very strong rule PVS1 is applied for a null variant located in a gene for which loss-of-function (LOF) is a known mechanism of disease. In practice, many tools interpret this rule by constructing a list of genes for which there are certain number of earlier identified pathogenic null variants. Benign rule, BP1 missense variant in a gene where only truncating variants cause disease, may reduce the importance of new pathogenic missense variant findings because in many cases disease associated genes have been found by identifying variants with clear, big impact (truncating/null variants).</p>
<p>It’s important to be aware of the potential caveats and limitations of the classification system, especially when identifying pathogenic variants in a clinical setting. ACMG system should work as a guideline to restrict the number of variants of interest for further inspection and not completely erase the manual work required to assess the importance of variants on patient level.</p>
<p>More work should still be done to clarify and develop the guidelines in order to reduce the inconsistencies.  This is important also for the machine learning based methods because many tools use the variant databases, consisting of variants with manual or automated ACMG classifications, as training data.</p>
<h2 id="references">References</h2>
<p>Richards S, Aziz N, Bale S, et al. Standards and guidelines for the interpretation of sequence variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology. <em>Genet Med</em>. 17(5):405-424. (2015) doi:10.1038/gim.2015.30</p>
<p>Karczewski KJ., Francioli LC., Tiao, G. et al. The mutational constraint spectrum quantified from variation in 141,456 humans. <em>Nature</em> ****581, 434–443 (2020). <a href="https://doi.org/10.1038/s41586-020-2308-7">https://doi.org/10.1038/s41586-020-2308-7</a></p>
<p>Cheng J et al. Accurate proteome-wide missense variant effect prediction with AlphaMissense. Science 381,eadg7492(2023). DOI:10.1126/science.adg7492</p>
<p>Landrum MJ, Lee JM, Riley GR, Jang W, Rubinstein WS, Church DM, Maglott DR. ClinVar: public archive of relationships among sequence variation and human phenotype. Nucleic Acids Res. 42(1):D980-5 (2014). doi: 10.1093/nar/gkt1113</p>
<p>Xavier A, Scott RJ, Talseth-Palmer BA. TAPES: A tool for assessment and prioritisation in exome studies. PLoS Comput Biol 15(10):e1007453 (2019). doi: 10.1371/journal.pcbi.1007453</p>
<p>Nicora, G., Zucca, S., Limongelli, I. et al. A machine learning approach based on ACMG/AMP guidelines for genomic variant classification and prioritization. Sci Rep 12, 2517 (2022). <a href="https://doi.org/10.1038/s41598-022-06547-3">https://doi.org/10.1038/s41598-022-06547-3</a></p>
<p>Kim Y, Ki C, Jang M.  Challenges and Considerations in Sequence Variant Interpretation for Mendelian Disorders.  Ann Lab Med 39:421-429 (2019).  https://doi.org/10.3343/alm.2019.39.5.421</p>


        


    </article>
</main>

    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="http://localhost:1313/">Cancer Data Diaries</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
<script>
    document.querySelectorAll('pre > code').forEach((codeblock) => {
        const container = codeblock.parentNode.parentNode;

        const copybutton = document.createElement('button');
        copybutton.classList.add('copy-code');
        copybutton.innerHTML = 'copy';

        function copyingDone() {
            copybutton.innerHTML = 'copied!';
            setTimeout(() => {
                copybutton.innerHTML = 'copy';
            }, 2000);
        }

        copybutton.addEventListener('click', (cb) => {
            if ('clipboard' in navigator) {
                navigator.clipboard.writeText(codeblock.textContent);
                copyingDone();
                return;
            }

            const range = document.createRange();
            range.selectNodeContents(codeblock);
            const selection = window.getSelection();
            selection.removeAllRanges();
            selection.addRange(range);
            try {
                document.execCommand('copy');
                copyingDone();
            } catch (e) { };
            selection.removeRange(range);
        });

        if (container.classList.contains("highlight")) {
            container.appendChild(copybutton);
        } else if (container.parentNode.firstChild == container) {
            
        } else if (codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName == "TABLE") {
            
            codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(copybutton);
        } else {
            
            codeblock.parentNode.appendChild(copybutton);
        }
    });
</script>
</body>

</html>
