<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>Creating mutated protein sequences from a list of amino acid changes | Cancer Data Diaries</title>
<meta name="keywords" content="ensembl, protein">
<meta name="description" content="In silico protein folding
Recent improvements in protein folding tools such as AlphaFold2
have dramatically changed protein structure research. In silico predictions have been able to reach comparable accuracy to experimental techniques such as X-ray crystallography and cryo-electron microscopy.
These tools may be useful to help filter protein structures for experimental studies.
Missense mutations
Missense mutations are single nucleotide changes in nucleotide sequence that lead to amino acid change in the
protein sequence. All single nucleotide changes don&rsquo;t lead to amino acid change, but may result in synonymous mutation
(no amino acid change), early stop codon (translation terminates) or a splicing change if located at splicing site.">
<meta name="author" content="Me">
<link rel="canonical" href="http://localhost:1313/posts/mutating-protein-sequence/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.b648782d138748285f2b1210a71b68efe4fa23d8cf11526df0b2fb74770c0de8.css" integrity="sha256-tkh4LROHSChfKxIQpxto7&#43;T6I9jPEVJt8LL7dHcMDeg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" hreflang="en" href="http://localhost:1313/posts/mutating-protein-sequence/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript><meta property="og:title" content="Creating mutated protein sequences from a list of amino acid changes" />
<meta property="og:description" content="In silico protein folding
Recent improvements in protein folding tools such as AlphaFold2
have dramatically changed protein structure research. In silico predictions have been able to reach comparable accuracy to experimental techniques such as X-ray crystallography and cryo-electron microscopy.
These tools may be useful to help filter protein structures for experimental studies.
Missense mutations
Missense mutations are single nucleotide changes in nucleotide sequence that lead to amino acid change in the
protein sequence. All single nucleotide changes don&rsquo;t lead to amino acid change, but may result in synonymous mutation
(no amino acid change), early stop codon (translation terminates) or a splicing change if located at splicing site." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/posts/mutating-protein-sequence/" />
<meta property="og:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta property="article:section" content="posts" />
<meta property="article:published_time" content="2024-08-02T16:20:16+09:00" />
<meta property="article:modified_time" content="2024-08-02T16:20:16+09:00" /><meta property="og:site_name" content="Cancer Data Diaries" />


<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta name="twitter:title" content="Creating mutated protein sequences from a list of amino acid changes"/>
<meta name="twitter:description" content="In silico protein folding
Recent improvements in protein folding tools such as AlphaFold2
have dramatically changed protein structure research. In silico predictions have been able to reach comparable accuracy to experimental techniques such as X-ray crystallography and cryo-electron microscopy.
These tools may be useful to help filter protein structures for experimental studies.
Missense mutations
Missense mutations are single nucleotide changes in nucleotide sequence that lead to amino acid change in the
protein sequence. All single nucleotide changes don&rsquo;t lead to amino acid change, but may result in synonymous mutation
(no amino acid change), early stop codon (translation terminates) or a splicing change if located at splicing site."/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position":  1 ,
      "name": "Posts",
      "item": "http://localhost:1313/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "Creating mutated protein sequences from a list of amino acid changes",
      "item": "http://localhost:1313/posts/mutating-protein-sequence/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "Creating mutated protein sequences from a list of amino acid changes",
  "name": "Creating mutated protein sequences from a list of amino acid changes",
  "description": "In silico protein folding Recent improvements in protein folding tools such as AlphaFold2 have dramatically changed protein structure research. In silico predictions have been able to reach comparable accuracy to experimental techniques such as X-ray crystallography and cryo-electron microscopy. These tools may be useful to help filter protein structures for experimental studies.\nMissense mutations Missense mutations are single nucleotide changes in nucleotide sequence that lead to amino acid change in the protein sequence. All single nucleotide changes don\u0026rsquo;t lead to amino acid change, but may result in synonymous mutation (no amino acid change), early stop codon (translation terminates) or a splicing change if located at splicing site.\n",
  "keywords": [
    "ensembl", "protein"
  ],
  "articleBody": "In silico protein folding Recent improvements in protein folding tools such as AlphaFold2 have dramatically changed protein structure research. In silico predictions have been able to reach comparable accuracy to experimental techniques such as X-ray crystallography and cryo-electron microscopy. These tools may be useful to help filter protein structures for experimental studies.\nMissense mutations Missense mutations are single nucleotide changes in nucleotide sequence that lead to amino acid change in the protein sequence. All single nucleotide changes don’t lead to amino acid change, but may result in synonymous mutation (no amino acid change), early stop codon (translation terminates) or a splicing change if located at splicing site.\nSingle nucleotide changes The effect of an amino acid change on protein structure and function is not always clear when only looking at the amino acid sequence. Some changes occur at known active sites or post-translational modification sites which may strongly indicate a deleterious effect on the protein function.\nBut what about the changes located outside of these sites?\nTraditionally the effect of the amino acid changes has been studied experimentally: producing protein products with the mutated amino acid, extracting the protein and solving its protein structure with for example X-ray crystallography. Experimental protocols are expensive and time-consuming and hence there’s a huge interest in pre-filtering the potential structures in cases where there are a large number of mutations of interest.\nCreating mutated sequences for in silico analysis I recently wanted to automate mutating protein FASTA sequences for AlphaFold2 protein folding and made a python script for that.\nSet API connection The script uses Ensembl REST API GRCh37 version and first sets a connection to the API:\n1import requests 2from requests.adapters import HTTPAdapter 3from requests.packages.urllib3.util.retry import Retry 4from requests_toolbelt import sessions 5 6 7def set_connection(): 8\tserver = \"http://grch37.rest.ensembl.org\" 9\tsession = sessions.BaseUrlSession(base_url = server) 10 11\tretry_strategy = Retry( 12\ttotal=3, 13\tbackoff_factor=2, 14\tstatus_forcelist=[429,500,502,503,504], 15\tmethod_whitelist=['HEAD','GET'] 16\t) 17 18\tadapter = HTTPAdapter(max_retries=retry_strategy) 19\tsession.mount('http://',adapter) 20 21\treturn session 22 23 24session = set_connection() This creates a lasting connection to the API during which you can do multiple requests without closing and opening the connection each time.\nRead in data After the connection is set you read in the data, which is a text file with column “Protein_change” and one amino acid change of the format A256T per row (All changes for the SAME protein):\nProtein_change A356T V14F 1def read_data(data): 2 try: 3 datafile = pd.read_csv(data, sep='\\t') 4 except FileNotFoundError: 5 logging.exception(\"Data file not found\") 6 sys.exit() 7 else: 8 logging.info(f\"Datafile {data} read successfully.\") 9 return datafile 10 11data = read_data(File) Ideally you have all your amino acid changes for the same protein transcript but that is not always the case when gathering mutations from literature. Some proteins may have differing amino acid transcripts; longer/shorter versions, or older/newer versions that have a bit differing mutation locations. This script currently takes the default/most common protein sequence from Ensembl server based on Ensembl transcript ENST and checks if the location and amino acid matches. If the mutation belongs to different protein transcript, it is skipped.\nGet protein FASTA sequence First request the protein sequence (ENSP) from the API, and separate the ENSP name and amino acid sequence in their own variables:\n1def get_protein_seq(ENST,session): 2 ext = f\"/sequence/id/{ENST}?type=protein\" 3 4 p_seq = request_server(ext, session) 5 ESP,seq = p_seq.text.split('\\n',1) 6 seq = seq.replace('\\n','') 7 logging.info(f\"Read protein sequence {ESP} for transcript {ENST}.\") 8 return(ESP,seq) 9 10ENSP, seq = get_protein_seq(ENST,session) Make amino acid changes to the sequence Next, make an amino acid change to the amino acid sequence for the location specified in the ‘Protein_change’ column of the data file. Make two lists, first containing the ENSP-code + amino acid change as a name for the sequence and the second one includes the mutated amino acid sequences.\n1def mutate_pseq(changes, ENSP, seq): 2 name_list = [] 3 seq_list = [] 4 5 name_list.append(f'{ENSP}_WT') 6 seq_list.append(seq) 7 8 9 for c in changes['Protein_change']: 10\t#unmutated sequence 11 temp = seq 12\t#position of mutation eg. A265T -\u003e 265 13 k = int(c[5:-3]) 14 15\t#checking if mutation is a missense mutation 16 if seq[k-1] == a_as[c[2:5]]: 17 temp2 = temp[:k-1] + a_as[c[-3:]] + temp[k:] 18 name_list.append(ENSP+\"_\"+c) 19 seq_list.append(temp2) 20 logging.info(f\"Mutated protein sequence with change {c}.\") 21 else: 22\t#if not missense mutation 23\t#or mutation not on default protein sequence 24 logging.info(f'Something wrong with {c}') 25 26 return(name_list, seq_list) 27 28nlist, slist = mutate_pseq(data, ENSP[1:-2], seq) Write results to a file Finally, write out the results as FASTA format to a single file:\nProtein FASTA format:\n\u003eENSP12344678_A345T ADGJRKAJGJRHJRJKDHEUIEJAHK\n1def save_seqs(nlist, slist, outname): 2 with open(f\"{outname}.fasta\",\"w\") as f: 3 for i in range(len(slist)): 4 f.write(nlist[i] + \"\\n\" + slist[i] + \"\\n\") 5 6 logging.info(f\"File {outname}.fasta saved.\") Summary / future to do’s Complete script can be found in github.\nFuture modifications to the script:\n[] Multiple isoform checks\n[] Add GRCh38 version API\n[] Make an option to write sequences to a single OR separate FASTA files\n",
  "wordCount" : "826",
  "inLanguage": "en",
  "image": "http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E","datePublished": "2024-08-02T16:20:16+09:00",
  "dateModified": "2024-08-02T16:20:16+09:00",
  "author":{
    "@type": "Person",
    "name": "Me"
  },
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://localhost:1313/posts/mutating-protein-sequence/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Cancer Data Diaries",
    "logo": {
      "@type": "ImageObject",
      "url": "http://localhost:1313/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Home (Alt + H)">
                <img src="http://localhost:1313/icon.png" alt="" aria-label="logo"
                    height="35">Home</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/categories/" title="Categories">
                    <span>Categories</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search/" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/tags/" title="Tags">
                    <span>Tags</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main">

<article class="post-single">
  <header class="post-header">
    <div class="breadcrumbs"><a href="http://localhost:1313/">Home</a>&nbsp;»&nbsp;<a href="http://localhost:1313/posts/">Posts</a></div>
    <h1 class="post-title entry-hint-parent">
      Creating mutated protein sequences from a list of amino acid changes
    </h1>
    <div class="post-meta"><span title='2024-08-02 16:20:16 +0900 +0900'>August 2, 2024</span>&nbsp;·&nbsp;4 min&nbsp;·&nbsp;Me&nbsp;|&nbsp;<a href="https://github.com/adityatelange/hugo-PaperMod/tree/exampleSite/content/posts/Mutating-protein-sequence/index.md" rel="noopener noreferrer" target="_blank">Suggest Changes</a>

</div>
  </header> <div class="toc">
    <details >
        <summary accesskey="c" title="(Alt + C)">
            <span class="details">Table of Contents</span>
        </summary>

        <div class="inner"><ul>
                <li>
                    <a href="#in-silico-protein-folding" aria-label="In silico protein folding">In silico protein folding</a></li>
                <li>
                    <a href="#missense-mutations" aria-label="Missense mutations">Missense mutations</a></li>
                <li>
                    <a href="#creating-mutated-sequences-for-in-silico-analysis" aria-label="Creating mutated sequences for in silico analysis">Creating mutated sequences for in silico analysis</a><ul>
                        <ul>
                        
                <li>
                    <a href="#set-api-connection" aria-label="Set API connection">Set API connection</a></li>
                <li>
                    <a href="#read-in-data" aria-label="Read in data">Read in data</a></li>
                <li>
                    <a href="#get-protein-fasta-sequence" aria-label="Get protein FASTA sequence">Get protein FASTA sequence</a></li>
                <li>
                    <a href="#make-amino-acid-changes-to-the-sequence" aria-label="Make amino acid changes to the sequence">Make amino acid changes to the sequence</a></li>
                <li>
                    <a href="#write-results-to-a-file" aria-label="Write results to a file">Write results to a file</a></li></ul>
                    </ul>
                </li>
                <li>
                    <a href="#summary--future-to-dos" aria-label="Summary / future to do&rsquo;s">Summary / future to do&rsquo;s</a>
                </li>
            </ul>
        </div>
    </details>
</div>

  <div class="post-content"><h1 id="in-silico-protein-folding"><em>In silico</em> protein folding<a hidden class="anchor" aria-hidden="true" href="#in-silico-protein-folding">#</a></h1>
<p>Recent improvements in protein <a href="https://github.com/biolists/folding_tools">folding tools</a> such as AlphaFold2
have dramatically changed protein structure research. <em>In silico</em> predictions have been able to reach comparable accuracy to experimental techniques such as X-ray crystallography and cryo-electron microscopy.
These tools may be useful to help filter protein structures for experimental studies.</p>
<h1 id="missense-mutations">Missense mutations<a hidden class="anchor" aria-hidden="true" href="#missense-mutations">#</a></h1>
<p>Missense mutations are single nucleotide changes in nucleotide sequence that lead to amino acid change in the
protein sequence. All single nucleotide changes don&rsquo;t lead to amino acid change, but may result in synonymous mutation
(no amino acid change), early stop codon (translation terminates) or a splicing change if located at splicing site.</p>
<figure>
    <img loading="lazy" src="images/nucleotide_changes.png"/> <figcaption>
            Single nucleotide changes
        </figcaption>
</figure>

<p>The effect of an amino acid change on protein structure and function is not always clear when only looking at the amino acid sequence.
Some changes occur at known active sites or post-translational modification sites which may strongly indicate a deleterious effect on the protein function.</p>
<p>But what about the changes located outside of these sites?</p>
<p>Traditionally the effect of the amino acid changes has been studied experimentally: producing protein products with the mutated amino acid,
extracting the protein and solving its protein structure with for example X-ray crystallography.
Experimental protocols are expensive and time-consuming and hence there&rsquo;s a huge interest in pre-filtering the potential structures in cases where there are a large number of mutations of interest.</p>
<h1 id="creating-mutated-sequences-for-in-silico-analysis">Creating mutated sequences for <em>in silico</em> analysis<a hidden class="anchor" aria-hidden="true" href="#creating-mutated-sequences-for-in-silico-analysis">#</a></h1>
<p>I recently wanted to automate mutating protein FASTA sequences for AlphaFold2 protein folding and made a python script for that.</p>
<h3 id="set-api-connection">Set API connection<a hidden class="anchor" aria-hidden="true" href="#set-api-connection">#</a></h3>
<p>The script uses Ensembl <a href="http://grch37.rest.ensembl.org/">REST API</a> GRCh37 version and first sets a connection to the API:</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="kn">import</span> <span class="nn">requests</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl"><span class="kn">from</span> <span class="nn">requests.adapters</span> <span class="kn">import</span> <span class="n">HTTPAdapter</span>
</span></span><span class="line"><span class="ln"> 3</span><span class="cl"><span class="kn">from</span> <span class="nn">requests.packages.urllib3.util.retry</span> <span class="kn">import</span> <span class="n">Retry</span>
</span></span><span class="line"><span class="ln"> 4</span><span class="cl"><span class="kn">from</span> <span class="nn">requests_toolbelt</span> <span class="kn">import</span> <span class="n">sessions</span>
</span></span><span class="line"><span class="ln"> 5</span><span class="cl">
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">
</span></span><span class="line"><span class="ln"> 7</span><span class="cl"><span class="k">def</span> <span class="nf">set_connection</span><span class="p">():</span>
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">	<span class="n">server</span> <span class="o">=</span> <span class="s2">&#34;http://grch37.rest.ensembl.org&#34;</span>
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">	<span class="n">session</span> <span class="o">=</span> <span class="n">sessions</span><span class="o">.</span><span class="n">BaseUrlSession</span><span class="p">(</span><span class="n">base_url</span> <span class="o">=</span> <span class="n">server</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">10</span><span class="cl">
</span></span><span class="line"><span class="ln">11</span><span class="cl">	<span class="n">retry_strategy</span> <span class="o">=</span> <span class="n">Retry</span><span class="p">(</span>
</span></span><span class="line"><span class="ln">12</span><span class="cl">		<span class="n">total</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">13</span><span class="cl">		<span class="n">backoff_factor</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">14</span><span class="cl">		<span class="n">status_forcelist</span><span class="o">=</span><span class="p">[</span><span class="mi">429</span><span class="p">,</span><span class="mi">500</span><span class="p">,</span><span class="mi">502</span><span class="p">,</span><span class="mi">503</span><span class="p">,</span><span class="mi">504</span><span class="p">],</span>
</span></span><span class="line"><span class="ln">15</span><span class="cl">		<span class="n">method_whitelist</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;HEAD&#39;</span><span class="p">,</span><span class="s1">&#39;GET&#39;</span><span class="p">]</span>
</span></span><span class="line"><span class="ln">16</span><span class="cl">	<span class="p">)</span>
</span></span><span class="line"><span class="ln">17</span><span class="cl">
</span></span><span class="line"><span class="ln">18</span><span class="cl">	<span class="n">adapter</span> <span class="o">=</span> <span class="n">HTTPAdapter</span><span class="p">(</span><span class="n">max_retries</span><span class="o">=</span><span class="n">retry_strategy</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">19</span><span class="cl">	<span class="n">session</span><span class="o">.</span><span class="n">mount</span><span class="p">(</span><span class="s1">&#39;http://&#39;</span><span class="p">,</span><span class="n">adapter</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">20</span><span class="cl">
</span></span><span class="line"><span class="ln">21</span><span class="cl">	<span class="k">return</span> <span class="n">session</span>
</span></span><span class="line"><span class="ln">22</span><span class="cl">
</span></span><span class="line"><span class="ln">23</span><span class="cl">
</span></span><span class="line"><span class="ln">24</span><span class="cl"><span class="n">session</span> <span class="o">=</span> <span class="n">set_connection</span><span class="p">()</span>
</span></span></code></pre></div><p>This creates a lasting connection to the API during which you can do multiple requests without closing and opening the connection each time.</p>
<h3 id="read-in-data">Read in data<a hidden class="anchor" aria-hidden="true" href="#read-in-data">#</a></h3>
<p>After the connection is set you read in the data, which is a text file with column &ldquo;Protein_change&rdquo; and one amino acid change of the format A256T per row (All changes for the SAME protein):</p>
<table>
  <thead>
      <tr>
          <th style="text-align: center">Protein_change</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: center">A356T</td>
      </tr>
      <tr>
          <td style="text-align: center">V14F</td>
      </tr>
  </tbody>
</table>
<div class="highlight my-class" id="my-codeblock2" height="600"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="k">def</span> <span class="nf">read_data</span><span class="p">(</span><span class="n">data</span><span class="p">):</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl">    <span class="k">try</span><span class="p">:</span>
</span></span><span class="line"><span class="ln"> 3</span><span class="cl">        <span class="n">datafile</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 4</span><span class="cl">    <span class="k">except</span> <span class="ne">FileNotFoundError</span><span class="p">:</span>
</span></span><span class="line"><span class="ln"> 5</span><span class="cl">        <span class="n">logging</span><span class="o">.</span><span class="n">exception</span><span class="p">(</span><span class="s2">&#34;Data file not found&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">        <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">()</span>
</span></span><span class="line"><span class="ln"> 7</span><span class="cl">    <span class="k">else</span><span class="p">:</span>
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">        <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&#34;Datafile </span><span class="si">{</span><span class="n">data</span><span class="si">}</span><span class="s2"> read successfully.&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">        <span class="k">return</span> <span class="n">datafile</span>
</span></span><span class="line"><span class="ln">10</span><span class="cl">
</span></span><span class="line"><span class="ln">11</span><span class="cl"><span class="n">data</span> <span class="o">=</span> <span class="n">read_data</span><span class="p">(</span><span class="n">File</span><span class="p">)</span>
</span></span></code></pre></div><p>Ideally you have all your amino acid changes for the same protein transcript but that is not always the case when gathering mutations from literature. Some proteins may have differing amino acid transcripts; longer/shorter versions, or older/newer versions that have a bit differing mutation locations.
This script currently takes the default/most common protein sequence from Ensembl server based on Ensembl transcript ENST and checks if the location and amino acid matches. If the mutation belongs to different protein transcript, it is skipped.</p>
<h3 id="get-protein-fasta-sequence">Get protein FASTA sequence<a hidden class="anchor" aria-hidden="true" href="#get-protein-fasta-sequence">#</a></h3>
<p>First request the protein sequence (ENSP) from the API, and separate the ENSP name and amino acid sequence in their own variables:</p>
<div class="highlight my-class" id="my-codeblock3" height="600"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="k">def</span> <span class="nf">get_protein_seq</span><span class="p">(</span><span class="n">ENST</span><span class="p">,</span><span class="n">session</span><span class="p">):</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl">    <span class="n">ext</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&#34;/sequence/id/</span><span class="si">{</span><span class="n">ENST</span><span class="si">}</span><span class="s2">?type=protein&#34;</span>
</span></span><span class="line"><span class="ln"> 3</span><span class="cl">
</span></span><span class="line"><span class="ln"> 4</span><span class="cl">    <span class="n">p_seq</span> <span class="o">=</span> <span class="n">request_server</span><span class="p">(</span><span class="n">ext</span><span class="p">,</span> <span class="n">session</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 5</span><span class="cl">    <span class="n">ESP</span><span class="p">,</span><span class="n">seq</span> <span class="o">=</span> <span class="n">p_seq</span><span class="o">.</span><span class="n">text</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">,</span><span class="mi">1</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">    <span class="n">seq</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 7</span><span class="cl">    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&#34;Read protein sequence </span><span class="si">{</span><span class="n">ESP</span><span class="si">}</span><span class="s2"> for transcript </span><span class="si">{</span><span class="n">ENST</span><span class="si">}</span><span class="s2">.&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">    <span class="k">return</span><span class="p">(</span><span class="n">ESP</span><span class="p">,</span><span class="n">seq</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">
</span></span><span class="line"><span class="ln">10</span><span class="cl"><span class="n">ENSP</span><span class="p">,</span> <span class="n">seq</span> <span class="o">=</span> <span class="n">get_protein_seq</span><span class="p">(</span><span class="n">ENST</span><span class="p">,</span><span class="n">session</span><span class="p">)</span>
</span></span></code></pre></div><h3 id="make-amino-acid-changes-to-the-sequence">Make amino acid changes to the sequence<a hidden class="anchor" aria-hidden="true" href="#make-amino-acid-changes-to-the-sequence">#</a></h3>
<p>Next, make an amino acid change to the amino acid sequence for the location specified in the &lsquo;Protein_change&rsquo; column of the data file.
Make two lists, first containing the ENSP-code + amino acid change as a name for the sequence and the second one includes the mutated amino acid sequences.</p>
<div class="highlight my-class" id="my-codeblock3" height="600"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="k">def</span> <span class="nf">mutate_pseq</span><span class="p">(</span><span class="n">changes</span><span class="p">,</span> <span class="n">ENSP</span><span class="p">,</span> <span class="n">seq</span><span class="p">):</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl">    <span class="n">name_list</span> <span class="o">=</span> <span class="p">[]</span>
</span></span><span class="line"><span class="ln"> 3</span><span class="cl">    <span class="n">seq_list</span> <span class="o">=</span> <span class="p">[]</span>
</span></span><span class="line"><span class="ln"> 4</span><span class="cl">
</span></span><span class="line"><span class="ln"> 5</span><span class="cl">    <span class="n">name_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">ENSP</span><span class="si">}</span><span class="s1">_WT&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">    <span class="n">seq_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 7</span><span class="cl">
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">    <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">changes</span><span class="p">[</span><span class="s1">&#39;Protein_change&#39;</span><span class="p">]:</span>
</span></span><span class="line"><span class="ln">10</span><span class="cl">	<span class="c1">#unmutated sequence</span>
</span></span><span class="line"><span class="ln">11</span><span class="cl">        <span class="n">temp</span> <span class="o">=</span> <span class="n">seq</span> 
</span></span><span class="line"><span class="ln">12</span><span class="cl">	<span class="c1">#position of mutation eg. A265T -&gt; 265</span>
</span></span><span class="line"><span class="ln">13</span><span class="cl">        <span class="n">k</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">c</span><span class="p">[</span><span class="mi">5</span><span class="p">:</span><span class="o">-</span><span class="mi">3</span><span class="p">])</span>
</span></span><span class="line"><span class="ln">14</span><span class="cl">
</span></span><span class="line"><span class="ln">15</span><span class="cl">	<span class="c1">#checking if mutation is a missense mutation</span>
</span></span><span class="line"><span class="ln">16</span><span class="cl">        <span class="k">if</span> <span class="n">seq</span><span class="p">[</span><span class="n">k</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">==</span> <span class="n">a_as</span><span class="p">[</span><span class="n">c</span><span class="p">[</span><span class="mi">2</span><span class="p">:</span><span class="mi">5</span><span class="p">]]:</span>
</span></span><span class="line"><span class="ln">17</span><span class="cl">            <span class="n">temp2</span> <span class="o">=</span> <span class="n">temp</span><span class="p">[:</span><span class="n">k</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">+</span> <span class="n">a_as</span><span class="p">[</span><span class="n">c</span><span class="p">[</span><span class="o">-</span><span class="mi">3</span><span class="p">:]]</span> <span class="o">+</span> <span class="n">temp</span><span class="p">[</span><span class="n">k</span><span class="p">:]</span>
</span></span><span class="line"><span class="ln">18</span><span class="cl">            <span class="n">name_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">ENSP</span><span class="o">+</span><span class="s2">&#34;_&#34;</span><span class="o">+</span><span class="n">c</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">19</span><span class="cl">            <span class="n">seq_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">temp2</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">20</span><span class="cl">            <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&#34;Mutated protein sequence with change </span><span class="si">{</span><span class="n">c</span><span class="si">}</span><span class="s2">.&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">21</span><span class="cl">        <span class="k">else</span><span class="p">:</span>
</span></span><span class="line"><span class="ln">22</span><span class="cl">	    <span class="c1">#if not missense mutation</span>
</span></span><span class="line"><span class="ln">23</span><span class="cl">	    <span class="c1">#or mutation not on default protein sequence</span>
</span></span><span class="line"><span class="ln">24</span><span class="cl">            <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Something wrong with </span><span class="si">{</span><span class="n">c</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">25</span><span class="cl">
</span></span><span class="line"><span class="ln">26</span><span class="cl">    <span class="k">return</span><span class="p">(</span><span class="n">name_list</span><span class="p">,</span> <span class="n">seq_list</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">27</span><span class="cl">
</span></span><span class="line"><span class="ln">28</span><span class="cl"><span class="n">nlist</span><span class="p">,</span> <span class="n">slist</span> <span class="o">=</span> <span class="n">mutate_pseq</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">ENSP</span><span class="p">[</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">2</span><span class="p">],</span> <span class="n">seq</span><span class="p">)</span>
</span></span></code></pre></div><h3 id="write-results-to-a-file">Write results to a file<a hidden class="anchor" aria-hidden="true" href="#write-results-to-a-file">#</a></h3>
<p>Finally, write out the results as FASTA format to a single file:</p>
<p><strong>Protein FASTA format:</strong></p>
<p>&gt;ENSP12344678_A345T <br>
ADGJRKAJGJRHJRJKDHEUIEJAHK</p>
<div class="highlight my-class" id="my-codeblock3" height="600"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="ln">1</span><span class="cl"><span class="k">def</span> <span class="nf">save_seqs</span><span class="p">(</span><span class="n">nlist</span><span class="p">,</span> <span class="n">slist</span><span class="p">,</span> <span class="n">outname</span><span class="p">):</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl">    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="sa">f</span><span class="s2">&#34;</span><span class="si">{</span><span class="n">outname</span><span class="si">}</span><span class="s2">.fasta&#34;</span><span class="p">,</span><span class="s2">&#34;w&#34;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
</span></span><span class="line"><span class="ln">3</span><span class="cl">        <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">slist</span><span class="p">)):</span>
</span></span><span class="line"><span class="ln">4</span><span class="cl">            <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">nlist</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">+</span> <span class="s2">&#34;</span><span class="se">\n</span><span class="s2">&#34;</span> <span class="o">+</span> <span class="n">slist</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">+</span> <span class="s2">&#34;</span><span class="se">\n</span><span class="s2">&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">5</span><span class="cl">
</span></span><span class="line"><span class="ln">6</span><span class="cl">    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&#34;File </span><span class="si">{</span><span class="n">outname</span><span class="si">}</span><span class="s2">.fasta saved.&#34;</span><span class="p">)</span>
</span></span></code></pre></div><h1 id="summary--future-to-dos">Summary / future to do&rsquo;s<a hidden class="anchor" aria-hidden="true" href="#summary--future-to-dos">#</a></h1>
<p>Complete script can be found in <a href="https://github.com/jezkoski/MutateProtein/tree/main">github</a>.</p>
<p><strong>Future modifications to the script:</strong></p>
<p>[] Multiple isoform checks</p>
<p>[] Add GRCh38 version API</p>
<p>[] Make an option to write sequences to a single OR separate FASTA files</p>


  </div>

  <footer class="post-footer">
    <ul class="post-tags">
      <li><a href="http://localhost:1313/tags/ensembl/">Ensembl</a></li>
      <li><a href="http://localhost:1313/tags/protein/">Protein</a></li>
    </ul>
<nav class="paginav">
  <a class="prev" href="http://localhost:1313/posts/germline-variant-classification/">
    <span class="title">« Prev</span>
    <br>
    <span>ACMG-AMP classification for germline variant pathogenicity</span>
  </a>
</nav>

  </footer>
</article>
    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="http://localhost:1313/">Cancer Data Diaries</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
<script>
    document.querySelectorAll('pre > code').forEach((codeblock) => {
        const container = codeblock.parentNode.parentNode;

        const copybutton = document.createElement('button');
        copybutton.classList.add('copy-code');
        copybutton.innerHTML = 'copy';

        function copyingDone() {
            copybutton.innerHTML = 'copied!';
            setTimeout(() => {
                copybutton.innerHTML = 'copy';
            }, 2000);
        }

        copybutton.addEventListener('click', (cb) => {
            if ('clipboard' in navigator) {
                navigator.clipboard.writeText(codeblock.textContent);
                copyingDone();
                return;
            }

            const range = document.createRange();
            range.selectNodeContents(codeblock);
            const selection = window.getSelection();
            selection.removeAllRanges();
            selection.addRange(range);
            try {
                document.execCommand('copy');
                copyingDone();
            } catch (e) { };
            selection.removeRange(range);
        });

        if (container.classList.contains("highlight")) {
            container.appendChild(copybutton);
        } else if (container.parentNode.firstChild == container) {
            
        } else if (codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName == "TABLE") {
            
            codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(copybutton);
        } else {
            
            codeblock.parentNode.appendChild(copybutton);
        }
    });
</script>
</body>

</html>
