<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37? | Cancer Data Diaries</title>
<meta name="keywords" content="reference genome">
<meta name="description" content="What is a reference genome?
A human reference genome, reference assembly, is a sequence of nucleotides representing genes
and non-coding regions of the genome.
It is made up from a combination multiple individual&rsquo;s DNA and hence not representing a
genome of any single human but an average of all humans in the world. However,
it is difficult to get proper representation of all humans due to the bias in the
selection of participants for the genome sequencing and the great variantion in some parts
of the genome between populations.
The first version was released in 2000 and there have been
multiple improvements and versions after that.">
<meta name="author" content="Me">
<link rel="canonical" href="http://localhost:1313/posts/human-reference-genomes/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.b648782d138748285f2b1210a71b68efe4fa23d8cf11526df0b2fb74770c0de8.css" integrity="sha256-tkh4LROHSChfKxIQpxto7&#43;T6I9jPEVJt8LL7dHcMDeg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" hreflang="en" href="http://localhost:1313/posts/human-reference-genomes/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript><meta property="og:title" content="Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37?" />
<meta property="og:description" content="What is a reference genome?
A human reference genome, reference assembly, is a sequence of nucleotides representing genes
and non-coding regions of the genome.
It is made up from a combination multiple individual&rsquo;s DNA and hence not representing a
genome of any single human but an average of all humans in the world. However,
it is difficult to get proper representation of all humans due to the bias in the
selection of participants for the genome sequencing and the great variantion in some parts
of the genome between populations.
The first version was released in 2000 and there have been
multiple improvements and versions after that." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/posts/human-reference-genomes/" />
<meta property="og:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta property="article:section" content="posts" />
<meta property="article:published_time" content="2024-08-03T15:30:11+09:00" />
<meta property="article:modified_time" content="2024-08-03T15:30:11+09:00" /><meta property="og:site_name" content="Cancer Data Diaries" />


<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta name="twitter:title" content="Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37?"/>
<meta name="twitter:description" content="What is a reference genome?
A human reference genome, reference assembly, is a sequence of nucleotides representing genes
and non-coding regions of the genome.
It is made up from a combination multiple individual&rsquo;s DNA and hence not representing a
genome of any single human but an average of all humans in the world. However,
it is difficult to get proper representation of all humans due to the bias in the
selection of participants for the genome sequencing and the great variantion in some parts
of the genome between populations.
The first version was released in 2000 and there have been
multiple improvements and versions after that."/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position":  1 ,
      "name": "Posts",
      "item": "http://localhost:1313/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37?",
      "item": "http://localhost:1313/posts/human-reference-genomes/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37?",
  "name": "Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37?",
  "description": "What is a reference genome? A human reference genome, reference assembly, is a sequence of nucleotides representing genes and non-coding regions of the genome. It is made up from a combination multiple individual\u0026rsquo;s DNA and hence not representing a genome of any single human but an average of all humans in the world. However, it is difficult to get proper representation of all humans due to the bias in the selection of participants for the genome sequencing and the great variantion in some parts of the genome between populations. The first version was released in 2000 and there have been multiple improvements and versions after that.\n",
  "keywords": [
    "reference genome"
  ],
  "articleBody": "What is a reference genome? A human reference genome, reference assembly, is a sequence of nucleotides representing genes and non-coding regions of the genome. It is made up from a combination multiple individual’s DNA and hence not representing a genome of any single human but an average of all humans in the world. However, it is difficult to get proper representation of all humans due to the bias in the selection of participants for the genome sequencing and the great variantion in some parts of the genome between populations. The first version was released in 2000 and there have been multiple improvements and versions after that.\nWhat is it used for? Reference genomes are used in genetic research as a reference map to help assemble sequencing data. They are also used in sample comparison to identify differences in the genome between individuals.\nWhat are the different versions? Institutional differences There are number of reference genome assemblies available from different sources which are then named based on the institution.\nReference Institute Contig names Resource Notes GRCh37, GRCh38 The Genome Reference Consortium (GRC) chr prefix ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.25_GRCh37.p13/GCF_000001405.25_GRCh37.p13_genomic.fna.gz Constructed from sequenced bacterial artificial chromosomes (BACs) hg19, hg38 The University of California at Santa Cruz (UCSC) chr prefix http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz Based on GRCh reference but contains some alterations b37 Broad institute no ‘chr’ prefix https://console.cloud.google.com/storage/browser/_details/broad-references/hg19/v0/Homo_sapiens_assembly19.fasta Based on GRCh. Alterations: bases with low confidence masked out. human_g1k_v37 The 1000 Genome project No ‘chr’ prefix http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz Equivalent to b37, but without human herpesvirus 4 type 1 decoy sequence Mostly the references contain the same data, although they use different naming for contigs (hg19 and GRCh use ‘chr’ prefix whereas other two don’t).\nHowever, there are some sequence differences between the references. There are differences on chromosomes 3 and Y, and mitochondrial contig as can be seen on the reference comparison table at broad institutes article linked below.\nRelease builds 37 vs 38 Each major version of human reference genome assemblies are numbered differently and the latest ones are GRCh37 (release 2009) and GRCh38 (release 2013) (hg19 and hg38 in the UCSC). Each version comes with updated annotations to the earlier.\nGRCh38 assembly has repaired incorrect reads and misassembled areas in GRCh37. In addition, it has been updated to include model centromere sequences and alternate loci. GRCh38 has greater amoun of ALT contigs, which are pieces of genome that represent alternative haplotypes.\nReferences: https://www.science.org/doi/10.1126/science.abj6987\nhttps://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies?id=23390\nhttps://www.ncbi.nlm.nih.gov/grc\n",
  "wordCount" : "397",
  "inLanguage": "en",
  "image": "http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E","datePublished": "2024-08-03T15:30:11+09:00",
  "dateModified": "2024-08-03T15:30:11+09:00",
  "author":{
    "@type": "Person",
    "name": "Me"
  },
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://localhost:1313/posts/human-reference-genomes/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Cancer Data Diaries",
    "logo": {
      "@type": "ImageObject",
      "url": "http://localhost:1313/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Home (Alt + H)">
                <img src="http://localhost:1313/icon.png" alt="" aria-label="logo"
                    height="35">Home</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/categories/" title="Categories">
                    <span>Categories</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search/" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/tags/" title="Tags">
                    <span>Tags</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main">

<article class="post-single">
  <header class="post-header">
    <div class="breadcrumbs"><a href="http://localhost:1313/">Home</a>&nbsp;»&nbsp;<a href="http://localhost:1313/posts/">Posts</a></div>
    <h1 class="post-title entry-hint-parent">
      Human Reference Genomes: GRCh37, GRCh38, hg19, hg38, v37, b37?
    </h1>
    <div class="post-meta"><span title='2024-08-03 15:30:11 +0900 +0900'>August 3, 2024</span>&nbsp;·&nbsp;2 min&nbsp;·&nbsp;Me&nbsp;|&nbsp;<a href="https://github.com/adityatelange/hugo-PaperMod/tree/exampleSite/content/posts/Human-reference-genomes/index.md" rel="noopener noreferrer" target="_blank">Suggest Changes</a>

</div>
  </header> <div class="toc">
    <details >
        <summary accesskey="c" title="(Alt + C)">
            <span class="details">Table of Contents</span>
        </summary>

        <div class="inner"><ul>
                <li>
                    <a href="#what-is-a-reference-genome" aria-label="What is a reference genome?">What is a reference genome?</a></li>
                <li>
                    <a href="#what-is-it-used-for" aria-label="What is it used for?">What is it used for?</a></li>
                <li>
                    <a href="#what-are-the-different-versions" aria-label="What are the different versions?">What are the different versions?</a><ul>
                        
                <li>
                    <a href="#institutional-differences" aria-label="Institutional differences">Institutional differences</a></li>
                <li>
                    <a href="#release-builds-37-vs-38" aria-label="Release builds 37 vs 38">Release builds 37 vs 38</a></li>
                <li>
                    <a href="#references" aria-label="References:">References:</a>
                </li>
            </ul>
            </li>
            </ul>
        </div>
    </details>
</div>

  <div class="post-content"><h2 id="what-is-a-reference-genome">What is a reference genome?<a hidden class="anchor" aria-hidden="true" href="#what-is-a-reference-genome">#</a></h2>
<p>A human reference genome, reference assembly, is a sequence of nucleotides representing genes
and non-coding regions of the genome.
It is made up from a combination multiple individual&rsquo;s DNA and hence not representing a
genome of any single human but an average of all humans in the world. However,
it is difficult to get proper representation of all humans due to the bias in the
selection of participants for the genome sequencing and the great variantion in some parts
of the genome between populations.
The first version was released in 2000 and there have been
multiple improvements and versions after that.</p>
<h2 id="what-is-it-used-for">What is it used for?<a hidden class="anchor" aria-hidden="true" href="#what-is-it-used-for">#</a></h2>
<p>Reference genomes are used in genetic research as a reference map to help assemble
sequencing data. They are also used in sample comparison to identify
differences in the genome between individuals.</p>
<h2 id="what-are-the-different-versions">What are the different versions?<a hidden class="anchor" aria-hidden="true" href="#what-are-the-different-versions">#</a></h2>
<h3 id="institutional-differences">Institutional differences<a hidden class="anchor" aria-hidden="true" href="#institutional-differences">#</a></h3>
<p>There are number of reference genome assemblies available from different sources which
are then named based on the institution.</p>
<table>
  <thead>
      <tr>
          <th style="text-align: center">Reference</th>
          <th style="text-align: center">Institute</th>
          <th style="text-align: center">Contig names</th>
          <th style="text-align: center">Resource</th>
          <th style="text-align: center">Notes</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: center">GRCh37, GRCh38</td>
          <td style="text-align: center">The Genome Reference Consortium (GRC)</td>
          <td style="text-align: center">chr prefix</td>
          <td style="text-align: center"><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.25_GRCh37.p13/GCF_000001405.25_GRCh37.p13_genomic.fna.gz">ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.25_GRCh37.p13/GCF_000001405.25_GRCh37.p13_genomic.fna.gz</a></td>
          <td style="text-align: center">Constructed from sequenced bacterial artificial chromosomes (BACs)</td>
      </tr>
      <tr>
          <td style="text-align: center">hg19, hg38</td>
          <td style="text-align: center">The University of California at Santa Cruz (UCSC)</td>
          <td style="text-align: center">chr prefix</td>
          <td style="text-align: center"><a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.fa.gz</a></td>
          <td style="text-align: center">Based on GRCh reference but contains some alterations</td>
      </tr>
      <tr>
          <td style="text-align: center">b37</td>
          <td style="text-align: center">Broad institute</td>
          <td style="text-align: center">no &lsquo;chr&rsquo; prefix</td>
          <td style="text-align: center"><a href="https://console.cloud.google.com/storage/browser/_details/broad-references/hg19/v0/Homo_sapiens_assembly19.fasta">https://console.cloud.google.com/storage/browser/_details/broad-references/hg19/v0/Homo_sapiens_assembly19.fasta</a></td>
          <td style="text-align: center">Based on GRCh. Alterations: bases with low confidence masked out.</td>
      </tr>
      <tr>
          <td style="text-align: center">human_g1k_v37</td>
          <td style="text-align: center">The 1000 Genome project</td>
          <td style="text-align: center">No &lsquo;chr&rsquo; prefix</td>
          <td style="text-align: center"><a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz">http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz</a></td>
          <td style="text-align: center">Equivalent to b37, but without human herpesvirus 4 type 1 decoy sequence</td>
      </tr>
  </tbody>
</table>
<p>Mostly the references contain the same data, although they use different naming for
contigs (hg19 and GRCh use ‘chr’ prefix whereas other two don’t).</p>
<p>However, there are some sequence differences between the references.
There are differences on chromosomes 3 and Y, and mitochondrial contig as can be seen
on the reference comparison table at broad institutes article linked below.</p>
<h3 id="release-builds-37-vs-38">Release builds 37 vs 38<a hidden class="anchor" aria-hidden="true" href="#release-builds-37-vs-38">#</a></h3>
<p>Each major version of human reference genome assemblies are numbered differently and the
latest ones are GRCh37 (release 2009) and GRCh38 (release 2013) (hg19 and hg38 in the UCSC). Each version comes with
updated annotations to the earlier.</p>
<p>GRCh38 assembly has repaired incorrect reads and misassembled areas in GRCh37.
In addition, it has been updated to include model centromere sequences and alternate
loci.  GRCh38 has greater amoun of ALT contigs, which are pieces of genome
that represent alternative haplotypes.</p>
<h3 id="references">References:<a hidden class="anchor" aria-hidden="true" href="#references">#</a></h3>
<p><a href="https://www.science.org/doi/10.1126/science.abj6987">https://www.science.org/doi/10.1126/science.abj6987</a></p>
<p><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies?id=23390">https://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies?id=23390</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/grc">https://www.ncbi.nlm.nih.gov/grc</a></p>


  </div>

  <footer class="post-footer">
    <ul class="post-tags">
      <li><a href="http://localhost:1313/tags/reference-genome/">Reference Genome</a></li>
    </ul>
<nav class="paginav">
  <a class="prev" href="http://localhost:1313/posts/make-maf-for-oncoplot/">
    <span class="title">« Prev</span>
    <br>
    <span>Make Maf for Oncoplot</span>
  </a>
  <a class="next" href="http://localhost:1313/posts/germline-variant-classification/">
    <span class="title">Next »</span>
    <br>
    <span>ACMG-AMP classification for germline variant pathogenicity</span>
  </a>
</nav>

  </footer>
</article>
    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="http://localhost:1313/">Cancer Data Diaries</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
<script>
    document.querySelectorAll('pre > code').forEach((codeblock) => {
        const container = codeblock.parentNode.parentNode;

        const copybutton = document.createElement('button');
        copybutton.classList.add('copy-code');
        copybutton.innerHTML = 'copy';

        function copyingDone() {
            copybutton.innerHTML = 'copied!';
            setTimeout(() => {
                copybutton.innerHTML = 'copy';
            }, 2000);
        }

        copybutton.addEventListener('click', (cb) => {
            if ('clipboard' in navigator) {
                navigator.clipboard.writeText(codeblock.textContent);
                copyingDone();
                return;
            }

            const range = document.createRange();
            range.selectNodeContents(codeblock);
            const selection = window.getSelection();
            selection.removeAllRanges();
            selection.addRange(range);
            try {
                document.execCommand('copy');
                copyingDone();
            } catch (e) { };
            selection.removeRange(range);
        });

        if (container.classList.contains("highlight")) {
            container.appendChild(copybutton);
        } else if (container.parentNode.firstChild == container) {
            
        } else if (codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName == "TABLE") {
            
            codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(copybutton);
        } else {
            
            codeblock.parentNode.appendChild(copybutton);
        }
    });
</script>
</body>

</html>
