<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>Make Maf for Oncoplot | Cancer Data Diaries</title>
<meta name="keywords" content="oncoplot, maftools, variant annotation">
<meta name="description" content="Check all articles in this list.
MAF files Mutation Annotation Format (MAF) is a data format used to store mutation data. Some visualization tools such as Bioconductor&rsquo;s oncoplot, require the data to be in MAF format.
MAF files have mandatory and optional fields. Mandatory fields:
Hugo_Symbol: Gene name (it&rsquo;s Hugo nomenclature name) Chromosome Start_Position End_Position Reference_allele Tumor_Seq_Allele2: alternative allele Variant_Classification: variant consequence eg. missense, stop, splice Variant_Type: single nucleotide variant, insertion, deletion Tumor_Sample_Barcode:	Tumor/Patient/Sample name or number Optional fields:">
<meta name="author" content="Me">
<link rel="canonical" href="http://localhost:1313/posts/make-maf-for-oncoplot/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.b648782d138748285f2b1210a71b68efe4fa23d8cf11526df0b2fb74770c0de8.css" integrity="sha256-tkh4LROHSChfKxIQpxto7&#43;T6I9jPEVJt8LL7dHcMDeg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" hreflang="en" href="http://localhost:1313/posts/make-maf-for-oncoplot/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript>
  

<meta property="og:title" content="Make Maf for Oncoplot" />
<meta property="og:description" content="Check all articles in this list.
MAF files Mutation Annotation Format (MAF) is a data format used to store mutation data. Some visualization tools such as Bioconductor&rsquo;s oncoplot, require the data to be in MAF format.
MAF files have mandatory and optional fields. Mandatory fields:
Hugo_Symbol: Gene name (it&rsquo;s Hugo nomenclature name) Chromosome Start_Position End_Position Reference_allele Tumor_Seq_Allele2: alternative allele Variant_Classification: variant consequence eg. missense, stop, splice Variant_Type: single nucleotide variant, insertion, deletion Tumor_Sample_Barcode:	Tumor/Patient/Sample name or number Optional fields:" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/posts/make-maf-for-oncoplot/" />
<meta property="og:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta property="article:section" content="posts" />
<meta property="article:published_time" content="2024-08-10T14:23:30+09:00" />
<meta property="article:modified_time" content="2024-08-10T14:23:30+09:00" /><meta property="og:site_name" content="Cancer Data Diaries" />



<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta name="twitter:title" content="Make Maf for Oncoplot"/>
<meta name="twitter:description" content="Check all articles in this list.
MAF files Mutation Annotation Format (MAF) is a data format used to store mutation data. Some visualization tools such as Bioconductor&rsquo;s oncoplot, require the data to be in MAF format.
MAF files have mandatory and optional fields. Mandatory fields:
Hugo_Symbol: Gene name (it&rsquo;s Hugo nomenclature name) Chromosome Start_Position End_Position Reference_allele Tumor_Seq_Allele2: alternative allele Variant_Classification: variant consequence eg. missense, stop, splice Variant_Type: single nucleotide variant, insertion, deletion Tumor_Sample_Barcode:	Tumor/Patient/Sample name or number Optional fields:"/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position":  1 ,
      "name": "Posts",
      "item": "http://localhost:1313/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "Make Maf for Oncoplot",
      "item": "http://localhost:1313/posts/make-maf-for-oncoplot/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "Make Maf for Oncoplot",
  "name": "Make Maf for Oncoplot",
  "description": "Check all articles in this list.\nMAF files Mutation Annotation Format (MAF) is a data format used to store mutation data. Some visualization tools such as Bioconductor\u0026rsquo;s oncoplot, require the data to be in MAF format.\nMAF files have mandatory and optional fields. Mandatory fields:\nHugo_Symbol: Gene name (it\u0026rsquo;s Hugo nomenclature name) Chromosome Start_Position End_Position Reference_allele Tumor_Seq_Allele2: alternative allele Variant_Classification: variant consequence eg. missense, stop, splice Variant_Type: single nucleotide variant, insertion, deletion Tumor_Sample_Barcode:\tTumor/Patient/Sample name or number Optional fields:",
  "keywords": [
    "oncoplot", "maftools", "variant annotation"
  ],
  "articleBody": "Check all articles in this list.\nMAF files Mutation Annotation Format (MAF) is a data format used to store mutation data. Some visualization tools such as Bioconductor’s oncoplot, require the data to be in MAF format.\nMAF files have mandatory and optional fields. Mandatory fields:\nHugo_Symbol: Gene name (it’s Hugo nomenclature name) Chromosome Start_Position End_Position Reference_allele Tumor_Seq_Allele2: alternative allele Variant_Classification: variant consequence eg. missense, stop, splice Variant_Type: single nucleotide variant, insertion, deletion Tumor_Sample_Barcode:\tTumor/Patient/Sample name or number Optional fields:\nnon MAF specific fields containing VAF and amino acid change information Generating MAF files When generating MAF files your starting point is usually a VCF file or a tabular file with or without annotations. Tool vcf2maf will annotate, prioritize transcripts, and generates a MAF suitable for further use. The tool requires VEP to be installed in order for it to be able to do the annotations for mutations.\nBecause I already had a VEP-annotated tabular file, I used python to generate MAF file and to change the values in Variant_Type and Variant_Classification columns to be maftools oncoplot compatible. More about bioconductor’s maftools can be found here.\nVEP annotated tabular file to MAF file Taking the required fields from the VEP-annotated tabular file and renaming the column names into oncoplot suitable ones.\n1import pandas as pd 2 3def make_maf(data): 4\tdata['End_Position'] = data['pos'] + 1 5\tmaf = data[['SYMBOL','chrom','pos','End_Position','ref','alt','Consequence','VARIANT_CLASS', 'patient']] 6\tmaf.columns = ['Hugo_Symbol', 'Chromosome','Start_Position','End_Position', 'Reference_Allele', 'Tumor_Seq_Allele2','Variant_Classification','Variant_Type','Tumor_Sample_Barcode'] 7\treturn maf 8 9maf = make_maf(data) Modify variant annotations In addition to VEP, there’s another popular variant annotation tool ANNOVAR. They use a bit different nomenclature on the variant consequence and variant type fields. Maftools oncoplot uses the ANNOVAR type of annotations and hence in R, I change the Variant_Type and Variant_Classification values from VEP nomenclature to ANNOVAR nomenclature.\n1library(dplyr) 2 3data \u003c- read.table('variants.maf', sep='\\t', header=TRUE) 4 5change_values \u003c- function(df) { 6\tdf \u003c- df %\u003e% 7\tmutate(Variant_Classification = case_when((Variant_Classification == 'missense_variant') ~ 'Missense_Mutation', 8\t(Variant_Classification == 'stop_gained') ~ 'Nonsense_Mutation', 9\t(Variant_Classification == 'splice_acceptor_variant') ~ 'Splice_Site', 10\t(Variant_Classification == 'splice_donor_variant') ~ 'Splice_Site', 11\t(Variant_Classification == 'missense_variant\u0026splice_region_variant') ~ 'Missense_Mutation', 12\t(Variant_Classification == 'frameshift_variant' \u0026 Variant_Type =='deletion') ~ 'Frame_Shift_Del', 13\t(Variant_Classification == 'start_lost') ~ 'Nonsense_Mutation', 14\t(Variant_Classification == 'frameshift_variant \u0026 Variant_Type == 'insertion') ~ 'Frame_Shift_Ins')) %\u003e% 15\tmutate(Variant_Type = case_when((Variant_Type == 'SNV') ~ 'SNP', 16\t(Variant_Type == 'insertion') ~ 'INS', 17\t(Variant_Type == 'deletion') ~ 'DEL')) 18 19\treturn(df) 20} 21 22data \u003c- change_values(data) Be careful with the variant nomenclature Each tool, VEP, ANNOVAR, and other annotation tools use different systems for variant nomenclature and they may vary. ANNOVAR annotations are more general compared to VEP. For example, in ANNOVAR, splice-site variants are annotated as ‘Splice_Site’, whereas in VEP they are annotated as ‘splice_acceptor_variant’ or ‘splice_donor_variant’ depending on their location. In addition, variants in multiple transcripts are dealt with differently. Some tools list each different annotation the variant gets in the different transcripts whereas for example, ANNOVAR outputs only the most deleterious variant based on their priorisation system.\nReferences VEP: https://asia.ensembl.org/info/docs/tools/vep/index.html ANNOVAR: https://annovar.openbioinformatics.org/en/latest/\n",
  "wordCount" : "497",
  "inLanguage": "en",
  "image": "http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E","datePublished": "2024-08-10T14:23:30+09:00",
  "dateModified": "2024-08-10T14:23:30+09:00",
  "author":{
    "@type": "Person",
    "name": "Me"
  },
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://localhost:1313/posts/make-maf-for-oncoplot/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Cancer Data Diaries",
    "logo": {
      "@type": "ImageObject",
      "url": "http://localhost:1313/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Home (Alt + H)">
                <img src="http://localhost:1313/icon.png" alt="" aria-label="logo"
                    height="35">Home</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/categories/" title="Categories">
                    <span>Categories</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search/" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/tags/" title="Tags">
                    <span>Tags</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main">
<main>
    <article>
    <header>
        <h1>Make Maf for Oncoplot</h1>
    </header>

        
<div>
    <h4>This article is part of a series.</h4>
    <ul class="list-group">
        
    </ul>
</div>



        <p><a href="http://localhost:1313/series/bio-visualisation-tools/">Check all articles in this list</a>.</p>
<h1 id="maf-files">MAF files</h1>
<p>Mutation Annotation Format (MAF) is a data format used to store mutation data.
Some visualization tools such as Bioconductor&rsquo;s oncoplot, require the data
to be in MAF format.</p>
<p>MAF files have mandatory and optional fields.
Mandatory fields:</p>
<ul>
<li>Hugo_Symbol: Gene name (it&rsquo;s Hugo nomenclature name)</li>
<li>Chromosome</li>
<li>Start_Position</li>
<li>End_Position</li>
<li>Reference_allele</li>
<li>Tumor_Seq_Allele2: alternative allele</li>
<li>Variant_Classification: variant consequence eg. missense, stop, splice</li>
<li>Variant_Type: single nucleotide variant, insertion, deletion</li>
<li>Tumor_Sample_Barcode:	Tumor/Patient/Sample name or number</li>
</ul>
<p>Optional fields:</p>
<ul>
<li>non MAF specific fields containing VAF and amino acid change information</li>
</ul>
<h1 id="generating-maf-files">Generating MAF files</h1>
<p>When generating MAF files your starting point is usually a VCF file or a tabular file with or without annotations.
Tool <a href="https://github.com/mskcc/vcf2maf">vcf2maf</a> will annotate, prioritize transcripts, and generates
a MAF suitable for further use. The tool requires VEP to be installed in order for it to be able to do the annotations for mutations.</p>
<p>Because I already had a VEP-annotated tabular file, I used python to generate
MAF file and to change the values in Variant_Type and Variant_Classification columns
to be maftools oncoplot compatible. More about bioconductor&rsquo;s maftools can be found <a href="https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html#2_Generating_MAF_files">here</a>.</p>
<h2 id="vep-annotated-tabular-file-to-maf-file">VEP annotated tabular file to MAF file</h2>
<p>Taking the required fields from the VEP-annotated tabular file and renaming the column names into oncoplot suitable ones.</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-python" data-lang="python"><span class="line"><span class="ln">1</span><span class="cl"><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl">
</span></span><span class="line"><span class="ln">3</span><span class="cl"><span class="k">def</span> <span class="nf">make_maf</span><span class="p">(</span><span class="n">data</span><span class="p">):</span>
</span></span><span class="line"><span class="ln">4</span><span class="cl">	<span class="n">data</span><span class="p">[</span><span class="s1">&#39;End_Position&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span>
</span></span><span class="line"><span class="ln">5</span><span class="cl">	<span class="n">maf</span> <span class="o">=</span> <span class="n">data</span><span class="p">[[</span><span class="s1">&#39;SYMBOL&#39;</span><span class="p">,</span><span class="s1">&#39;chrom&#39;</span><span class="p">,</span><span class="s1">&#39;pos&#39;</span><span class="p">,</span><span class="s1">&#39;End_Position&#39;</span><span class="p">,</span><span class="s1">&#39;ref&#39;</span><span class="p">,</span><span class="s1">&#39;alt&#39;</span><span class="p">,</span><span class="s1">&#39;Consequence&#39;</span><span class="p">,</span><span class="s1">&#39;VARIANT_CLASS&#39;</span><span class="p">,</span> <span class="s1">&#39;patient&#39;</span><span class="p">]]</span>
</span></span><span class="line"><span class="ln">6</span><span class="cl">	<span class="n">maf</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Hugo_Symbol&#39;</span><span class="p">,</span> <span class="s1">&#39;Chromosome&#39;</span><span class="p">,</span><span class="s1">&#39;Start_Position&#39;</span><span class="p">,</span><span class="s1">&#39;End_Position&#39;</span><span class="p">,</span> <span class="s1">&#39;Reference_Allele&#39;</span><span class="p">,</span> <span class="s1">&#39;Tumor_Seq_Allele2&#39;</span><span class="p">,</span><span class="s1">&#39;Variant_Classification&#39;</span><span class="p">,</span><span class="s1">&#39;Variant_Type&#39;</span><span class="p">,</span><span class="s1">&#39;Tumor_Sample_Barcode&#39;</span><span class="p">]</span>
</span></span><span class="line"><span class="ln">7</span><span class="cl">	<span class="k">return</span> <span class="n">maf</span>
</span></span><span class="line"><span class="ln">8</span><span class="cl">
</span></span><span class="line"><span class="ln">9</span><span class="cl"><span class="n">maf</span> <span class="o">=</span> <span class="n">make_maf</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</span></span></code></pre></div><h2 id="modify-variant-annotations">Modify variant annotations</h2>
<p>In addition to VEP, there&rsquo;s another popular variant annotation tool ANNOVAR. They use a bit
different nomenclature on the variant consequence and variant type fields.
Maftools oncoplot uses the ANNOVAR type of annotations and hence in R, I change the
Variant_Type and Variant_Classification values from VEP nomenclature to
ANNOVAR nomenclature.</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-R" data-lang="R"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl">
</span></span><span class="line"><span class="ln"> 3</span><span class="cl"><span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;variants.maf&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></span><span class="line"><span class="ln"> 4</span><span class="cl">
</span></span><span class="line"><span class="ln"> 5</span><span class="cl"><span class="n">change_values</span> <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="p">{</span>
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">	<span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span> <span class="o">%&gt;%</span>
</span></span><span class="line"><span class="ln"> 7</span><span class="cl">		<span class="nf">mutate</span><span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">=</span> <span class="nf">case_when</span><span class="p">((</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;missense_variant&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Missense_Mutation&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;stop_gained&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Nonsense_Mutation&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;splice_acceptor_variant&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Splice_Site&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">10</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;splice_donor_variant&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Splice_Site&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">11</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;missense_variant&amp;splice_region_variant&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Missense_Mutation&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">12</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;frameshift_variant&#39;</span> <span class="o">&amp;</span> <span class="n">Variant_Type</span> <span class="o">==</span><span class="s">&#39;deletion&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Frame_Shift_Del&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">13</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;start_lost&#39;</span><span class="p">)</span> <span class="o">~</span> <span class="s">&#39;Nonsense_Mutation&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">14</span><span class="cl">			<span class="p">(</span><span class="n">Variant_Classification</span> <span class="o">==</span> <span class="s">&#39;frameshift_variant &amp; Variant_Type == &#39;</span><span class="n">insertion</span><span class="s">&#39;) ~ &#39;</span><span class="n">Frame_Shift_Ins</span><span class="s">&#39;)) %&gt;%
</span></span></span><span class="line"><span class="ln">15</span><span class="cl"><span class="s">		mutate(Variant_Type = case_when((Variant_Type == &#39;</span><span class="n">SNV</span><span class="s">&#39;) ~ &#39;</span><span class="n">SNP</span><span class="s">&#39;,
</span></span></span><span class="line"><span class="ln">16</span><span class="cl"><span class="s">			(Variant_Type == &#39;</span><span class="n">insertion</span><span class="s">&#39;) ~ &#39;</span><span class="n">INS</span><span class="s">&#39;,
</span></span></span><span class="line"><span class="ln">17</span><span class="cl"><span class="s">			(Variant_Type == &#39;</span><span class="n">deletion</span><span class="s">&#39;) ~ &#39;</span><span class="n">DEL</span><span class="s">&#39;</span><span class="err">))</span>
</span></span><span class="line"><span class="ln">18</span><span class="cl">
</span></span><span class="line"><span class="ln">19</span><span class="cl">	<span class="kr">return</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">20</span><span class="cl"><span class="p">}</span>
</span></span><span class="line"><span class="ln">21</span><span class="cl">
</span></span><span class="line"><span class="ln">22</span><span class="cl"><span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">change_values</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</span></span></code></pre></div><h2 id="be-careful-with-the-variant-nomenclature">Be careful with the variant nomenclature</h2>
<p>Each tool, VEP, ANNOVAR, and other annotation tools use different systems for variant nomenclature
and they may vary. ANNOVAR annotations are more general compared to VEP. For example,
in ANNOVAR, splice-site variants are annotated as &lsquo;Splice_Site&rsquo;, whereas in VEP they are
annotated as &lsquo;splice_acceptor_variant&rsquo; or &lsquo;splice_donor_variant&rsquo; depending on their location.
In addition, variants in multiple transcripts are dealt with differently. Some tools list
each different annotation the variant gets in the different transcripts whereas for example,
ANNOVAR outputs only the most deleterious variant based on their priorisation system.</p>
<h1 id="references">References</h1>
<p>VEP: <a href="https://asia.ensembl.org/info/docs/tools/vep/index.html">https://asia.ensembl.org/info/docs/tools/vep/index.html</a>
ANNOVAR: <a href="https://annovar.openbioinformatics.org/en/latest/">https://annovar.openbioinformatics.org/en/latest/</a></p>


        
<div>
    <h4>This article is part of a series.</h4>
    <ul class="list-group">
        
    </ul>
</div>



    </article>
</main>

    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="http://localhost:1313/">Cancer Data Diaries</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
<script>
    document.querySelectorAll('pre > code').forEach((codeblock) => {
        const container = codeblock.parentNode.parentNode;

        const copybutton = document.createElement('button');
        copybutton.classList.add('copy-code');
        copybutton.innerHTML = 'copy';

        function copyingDone() {
            copybutton.innerHTML = 'copied!';
            setTimeout(() => {
                copybutton.innerHTML = 'copy';
            }, 2000);
        }

        copybutton.addEventListener('click', (cb) => {
            if ('clipboard' in navigator) {
                navigator.clipboard.writeText(codeblock.textContent);
                copyingDone();
                return;
            }

            const range = document.createRange();
            range.selectNodeContents(codeblock);
            const selection = window.getSelection();
            selection.removeAllRanges();
            selection.addRange(range);
            try {
                document.execCommand('copy');
                copyingDone();
            } catch (e) { };
            selection.removeRange(range);
        });

        if (container.classList.contains("highlight")) {
            container.appendChild(copybutton);
        } else if (container.parentNode.firstChild == container) {
            
        } else if (codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName == "TABLE") {
            
            codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(copybutton);
        } else {
            
            codeblock.parentNode.appendChild(copybutton);
        }
    });
</script>
</body>

</html>
