<!DOCTYPE html>
<html lang="en" dir="auto">

<head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script><meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="robots" content="index, follow">
<title>Oncoplot | Cancer Data Diaries</title>
<meta name="keywords" content="oncoplot, maftools">
<meta name="description" content="Introduction
Oncoplots are are visualisation tool used in genomics to present genetic mutations in a set of study samples.
They provide an overview of the mutation/variant types and frequencies across all the samples.
This allows for easier identification of recurrent and co-occuring mutations.
Each column in the oncoplot represent a sample or individual, and each row corresponds to a gene.
Mutation types are represented with different colors and symbols may be added to indicate extra information on the mutation.">
<meta name="author" content="Me">
<link rel="canonical" href="http://localhost:1313/posts/oncoplot/">
<link crossorigin="anonymous" href="/assets/css/stylesheet.b648782d138748285f2b1210a71b68efe4fa23d8cf11526df0b2fb74770c0de8.css" integrity="sha256-tkh4LROHSChfKxIQpxto7&#43;T6I9jPEVJt8LL7dHcMDeg=" rel="preload stylesheet" as="style">
<link rel="icon" href="http://localhost:1313/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/favicon-32x32.png">
<link rel="apple-touch-icon" href="http://localhost:1313/apple-touch-icon.png">
<link rel="mask-icon" href="http://localhost:1313/safari-pinned-tab.svg">
<meta name="theme-color" content="#2e2e33">
<meta name="msapplication-TileColor" content="#2e2e33">
<link rel="alternate" hreflang="en" href="http://localhost:1313/posts/oncoplot/">
<noscript>
    <style>
        #theme-toggle,
        .top-link {
            display: none;
        }

    </style>
    <style>
        @media (prefers-color-scheme: dark) {
            :root {
                --theme: rgb(29, 30, 32);
                --entry: rgb(46, 46, 51);
                --primary: rgb(218, 218, 219);
                --secondary: rgb(155, 156, 157);
                --tertiary: rgb(65, 66, 68);
                --content: rgb(196, 196, 197);
                --code-block-bg: rgb(46, 46, 51);
                --code-bg: rgb(55, 56, 62);
                --border: rgb(51, 51, 51);
            }

            .list {
                background: var(--theme);
            }

            .list:not(.dark)::-webkit-scrollbar-track {
                background: 0 0;
            }

            .list:not(.dark)::-webkit-scrollbar-thumb {
                border-color: var(--theme);
            }
        }

    </style>
</noscript><meta property="og:title" content="Oncoplot" />
<meta property="og:description" content="Introduction
Oncoplots are are visualisation tool used in genomics to present genetic mutations in a set of study samples.
They provide an overview of the mutation/variant types and frequencies across all the samples.
This allows for easier identification of recurrent and co-occuring mutations.
Each column in the oncoplot represent a sample or individual, and each row corresponds to a gene.
Mutation types are represented with different colors and symbols may be added to indicate extra information on the mutation." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/posts/oncoplot/" />
<meta property="og:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta property="article:section" content="posts" />
<meta property="article:published_time" content="2024-08-26T13:43:50+09:00" />
<meta property="article:modified_time" content="2024-08-26T13:43:50+09:00" /><meta property="og:site_name" content="Cancer Data Diaries" />



<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:image" content="http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E" />
<meta name="twitter:title" content="Oncoplot"/>
<meta name="twitter:description" content="Introduction
Oncoplots are are visualisation tool used in genomics to present genetic mutations in a set of study samples.
They provide an overview of the mutation/variant types and frequencies across all the samples.
This allows for easier identification of recurrent and co-occuring mutations.
Each column in the oncoplot represent a sample or individual, and each row corresponds to a gene.
Mutation types are represented with different colors and symbols may be added to indicate extra information on the mutation."/>


<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position":  1 ,
      "name": "Posts",
      "item": "http://localhost:1313/posts/"
    }, 
    {
      "@type": "ListItem",
      "position":  2 ,
      "name": "Oncoplot",
      "item": "http://localhost:1313/posts/oncoplot/"
    }
  ]
}
</script>
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "Oncoplot",
  "name": "Oncoplot",
  "description": "Introduction Oncoplots are are visualisation tool used in genomics to present genetic mutations in a set of study samples. They provide an overview of the mutation/variant types and frequencies across all the samples. This allows for easier identification of recurrent and co-occuring mutations.\nEach column in the oncoplot represent a sample or individual, and each row corresponds to a gene. Mutation types are represented with different colors and symbols may be added to indicate extra information on the mutation.\n",
  "keywords": [
    "oncoplot", "maftools"
  ],
  "articleBody": "Introduction Oncoplots are are visualisation tool used in genomics to present genetic mutations in a set of study samples. They provide an overview of the mutation/variant types and frequencies across all the samples. This allows for easier identification of recurrent and co-occuring mutations.\nEach column in the oncoplot represent a sample or individual, and each row corresponds to a gene. Mutation types are represented with different colors and symbols may be added to indicate extra information on the mutation.\nBasic oncoplot Oncoplots are often presented with additional information such as clinical parameters that can be added under, top, or to the sides of the oncoplot.\nThere are several tools for creating oncoplots and currently the most popular one is Bioconductor’s maftools oncoplot.\nWhy use oncoplots? Oncoplots are popular in cancer research because they make visualising complex data easier. Cancer genomes often carry a large number of mutations, which is why simplifying and visualising the most frequent ones is an important task.\nIn addition, oncoplot makes it easier to compare the mutation landscape between specific cancer types or therapeutic responses. This allows the identification of patterns associated with specific outcomes which can in turn gives new information about the cancer mechanisms or biomarkers for diagnosis.\nHow to create an oncoplot? Creating an oncoplot requires few steps from data preparation to visualisation.\nData requirements For this tutorial, you need genetic mutation data in MAF format. You can refer to this previous post on how to convert your annotated data into MAF format with a simple piece of code, or follow vcf2maf tutorial if you have VCF file.\nIf you want to add clinical information as metadata to the oncoplot, you need the clinical data in tabular format:\nTumor_Sample_Barcode Annotation1 Annotation2 Same sample codes as maf eg. diagnosis1 eg. sex Save the clinical data file with ending “.clin”\nTool requirements There are several tools and packages available for creating oncoplots. The most commonly used one is Bioconductor’s maftools oncoplot, a popular R package. For installation, run the following in R:\n1install.packages(\"BiocManager\") 2BiocManager::install(\"maftools\") In python, there are some options such as package fuc.\nCreating oncoplot with maftools oncoplot Read data Read in both MAF file with mutations and the optional clinical annotation file.\n1data \u003c- read.table(\"mutations.maf\", sep=\"\\t\", header=TRUE) 2clindata \u003c- read.table(\"clinicaldata.clin\", sep=\"\\t\", header=TRUE) Make any necessary filterings and changes to the file if needed. For example remember to modify VEP annotation mutation type nomenclature to an ANNOVAR types. Check the previous post on how to make a MAF file.\nCreate maftools dataobject Read files into maftools dataobject\n1laml \u003c- read.maf(maf = data, 2\tclinicalData = clindata, 3\tverbose = FALSE) 4 5getClinicalData(x = laml) Set annotation colors Maftools gives default colors for mutation types and other annotations, but if you want to change these, you need to declare the color separately. Also, if you have a large number of variables in some annotations, for example 15 unique diagnosis types, it will be wise to make your own color palette.\nIf you have a reasonable number of variables in an annotation, you can use RColorBrewer and the color palettes. Many of the RColorBrewer color palettes allow maximum of 12 colors. To display all the available color palettes in RColorBrewer run the following lines:\n1library(RColorBrewer) 2display.brewer.all() RColorBrewer palettes First retrieve a number of colors from a palette you want and then assign them to the annotations. Here’s an example how to annotate each mutation type their own colors from the ‘Paired’ palette.\n1vc_cols \u003c- RColorBrewer::brewer.pal(n=6, name='Paired') 2names(vc_cols) \u003c- c( 3\t'Frame_Shift_Del', 4\t'Missense_Mutation', 5\t'Nonsense_Mutation', 6\t'Multi_Hit', 7\t'Frame_shift_Ins', 8\t'Splice_Site' 9) In case of more than 12 colors, it is probably the best to construct a palette of your own by declaring all the colors of the palette separately. This is also a good way to construct a palette if you want some specific or different colors for some annotations.\nHere I constructed a list of distinct colors and assigned them to cancer diagnoses and sex.\n1dx_cols \u003c-c( 2\t\"dodgerblue2\", \"#E31A1C\", #red 3\t\"green4\", \"#6A3D9A\", # purple 4\t\"#FF7F00\", # orange 5 \"aquamarine3\", \"gold1\", 6 \"skyblue2\", \"deeppink1\", # lt pink 7 \"brown\", 8 \"#CAB2D6\", # lt purple 9 \"#FDBF6F\", # lt orange 10\t\"gray70\", \"khaki2\", 11 \"maroon\", \"orchid1\", \"blue4\", \"steelblue4\", 12 \"darkturquoise\", \"yellow4\", \"yellow3\", 13 \"darkorange4\" 14) 15 16names(dx_cols) \u003c- c( 17\t\"AML\", \"ALL\", \"Breast\", 18\t\"MM\", \"Colon\", \"Lung\",\"Skin\", 19\t\"BMF\",\"Ovary\",\"Lymphoma\", 20\t\"Pancreas\" 21) 22 23sex_cols \u003c- c(\"skyblue2\", \"orchid3\") 24names(sex_cols) \u003c- c(\"M\",\"F\") For annotations you need to save the colors into a list before making the oncoplot. Make sure that the list parameters (Dx_group and Sex in this case) are the same ones you have as column names in your clinical data annotation file!\n1anno_cols \u003c- list(Dx_group=dx_cols, Sex=sex_cols Make oncoplot with clinical annotations You have a lot of options how to modify an oncoplot and I suggest you to look into maftools oncoplot documentation to find out what additional things you can do with oncoplots. Here, we will draw a basic oncoplot with two different clinical annotations at the bottom of the plot.\n1oncoplot(maf = laml, #read in the maftools dataobject 2\tcolors = vc_cols, #colors for mutation types 3\tclinicalFeatures = c(\"Dx_group\", \"Sex\"), #the column names of features in the annotation file 4\tannotationColor = anno_cols, #list of colors for annotations 5\tremoveNonMutated = TRUE, #doesn't show samples if there are no mutations 6\tshowTitle = FALSE, 7\ttop = 10, #show only top 10 mutated genes 8\tshowPct = FALSE, #shows percentage of individuals carrying mutation in a gene 9\tdrawColBar = FALSE #draws a barplot on top to show how many mutations each individual have 10) And this gives a basic oncoplot with two clinical annotations. Sometimes you may need to adjust the legend parameters in order to make all the legends with (mutation type, Dx_group and Sex should all have their own legens underneath the oncoplot). Parameters to adjust for this are for example: legend_height, anno_height, gene_mar, annotationFontSize.\nBasic oncoplot References maftools oncoplot documentation vcf2maf documentation ",
  "wordCount" : "995",
  "inLanguage": "en",
  "image": "http://localhost:1313/%3Clink%20or%20path%20of%20image%20for%20opengraph,%20twitter-cards%3E","datePublished": "2024-08-26T13:43:50+09:00",
  "dateModified": "2024-08-26T13:43:50+09:00",
  "author":{
    "@type": "Person",
    "name": "Me"
  },
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://localhost:1313/posts/oncoplot/"
  },
  "publisher": {
    "@type": "Organization",
    "name": "Cancer Data Diaries",
    "logo": {
      "@type": "ImageObject",
      "url": "http://localhost:1313/favicon.ico"
    }
  }
}
</script>
</head>

<body class="" id="top">
<script>
    if (localStorage.getItem("pref-theme") === "dark") {
        document.body.classList.add('dark');
    } else if (localStorage.getItem("pref-theme") === "light") {
        document.body.classList.remove('dark')
    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
        document.body.classList.add('dark');
    }

</script>

<header class="header">
    <nav class="nav">
        <div class="logo">
            <a href="http://localhost:1313/" accesskey="h" title="Home (Alt + H)">
                <img src="http://localhost:1313/icon.png" alt="" aria-label="logo"
                    height="35">Home</a>
            <div class="logo-switches">
                <button id="theme-toggle" accesskey="t" title="(Alt + T)">
                    <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                    </svg>
                    <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24"
                        fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                        stroke-linejoin="round">
                        <circle cx="12" cy="12" r="5"></circle>
                        <line x1="12" y1="1" x2="12" y2="3"></line>
                        <line x1="12" y1="21" x2="12" y2="23"></line>
                        <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                        <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                        <line x1="1" y1="12" x2="3" y2="12"></line>
                        <line x1="21" y1="12" x2="23" y2="12"></line>
                        <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                        <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                    </svg>
                </button>
            </div>
        </div>
        <ul id="menu">
            <li>
                <a href="http://localhost:1313/archives" title="Archive">
                    <span>Archive</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/categories/" title="Categories">
                    <span>Categories</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/search/" title="Search (Alt &#43; /)" accesskey=/>
                    <span>Search</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/tags/" title="Tags">
                    <span>Tags</span>
                </a>
            </li>
            <li>
                <a href="http://localhost:1313/about" title="About">
                    <span>About</span>
                </a>
            </li>
        </ul>
    </nav>
</header>
<main class="main">

<article class="post-single">
  <header class="post-header">
    <div class="breadcrumbs"><a href="http://localhost:1313/">Home</a>&nbsp;»&nbsp;<a href="http://localhost:1313/posts/">Posts</a></div>
    <h1 class="post-title entry-hint-parent">
      Oncoplot
    </h1>
    <div class="post-meta"><span title='2024-08-26 13:43:50 +0900 +0900'>August 26, 2024</span>&nbsp;·&nbsp;5 min&nbsp;·&nbsp;Me&nbsp;|&nbsp;<a href="https://github.com/adityatelange/hugo-PaperMod/tree/exampleSite/content/posts/Oncoplot/index.md" rel="noopener noreferrer" target="_blank">Suggest Changes</a>

</div>
  </header> <div class="toc">
    <details >
        <summary accesskey="c" title="(Alt + C)">
            <span class="details">Table of Contents</span>
        </summary>

        <div class="inner"><ul>
                <li>
                    <a href="#introduction" aria-label="Introduction">Introduction</a></li>
                <li>
                    <a href="#why-use-oncoplots" aria-label="Why use oncoplots?">Why use oncoplots?</a></li>
                <li>
                    <a href="#how-to-create-an-oncoplot" aria-label="How to create an oncoplot?">How to create an oncoplot?</a><ul>
                        
                <li>
                    <a href="#data-requirements" aria-label="Data requirements">Data requirements</a></li>
                <li>
                    <a href="#tool-requirements" aria-label="Tool requirements">Tool requirements</a></li>
                <li>
                    <a href="#creating-oncoplot-with-maftools-oncoplot" aria-label="Creating oncoplot with maftools oncoplot">Creating oncoplot with maftools oncoplot</a><ul>
                        
                <li>
                    <a href="#read-data" aria-label="Read data">Read data</a></li>
                <li>
                    <a href="#create-maftools-dataobject" aria-label="Create maftools dataobject">Create maftools dataobject</a></li>
                <li>
                    <a href="#set-annotation-colors" aria-label="Set annotation colors">Set annotation colors</a></li>
                <li>
                    <a href="#make-oncoplot-with-clinical-annotations" aria-label="Make oncoplot with clinical annotations">Make oncoplot with clinical annotations</a></li></ul>
                </li></ul>
                </li>
                <li>
                    <a href="#references" aria-label="References">References</a>
                </li>
            </ul>
        </div>
    </details>
</div>

  <div class="post-content"><h1 id="introduction">Introduction<a hidden class="anchor" aria-hidden="true" href="#introduction">#</a></h1>
<p>Oncoplots are are visualisation tool used in genomics to present genetic mutations in a set of study samples.
They provide an overview of the mutation/variant types and frequencies across all the samples.
This allows for easier identification of recurrent and co-occuring mutations.</p>
<p>Each column in the oncoplot represent a sample or individual, and each row corresponds to a gene.
Mutation types are represented with different colors and symbols may be added to indicate extra information on the mutation.</p>
<figure>
    <img loading="lazy" src="oncoplot.jpg"/> <figcaption>
            Basic oncoplot
        </figcaption>
</figure>

<p>Oncoplots are often presented with additional information such as clinical
parameters that can be added under, top, or to the sides of the oncoplot.</p>
<p>There are several tools for creating oncoplots and currently the most
popular one is Bioconductor&rsquo;s maftools oncoplot.</p>
<h1 id="why-use-oncoplots">Why use oncoplots?<a hidden class="anchor" aria-hidden="true" href="#why-use-oncoplots">#</a></h1>
<p>Oncoplots are popular in cancer research because they make visualising complex data easier.
Cancer genomes often carry a large number of mutations, which is why simplifying and visualising
the most frequent ones is an important task.</p>
<p>In addition, oncoplot makes it easier to compare the mutation landscape between specific cancer types or
therapeutic responses. This allows the identification of patterns associated with specific outcomes which can
in turn gives new information about the cancer mechanisms or biomarkers for diagnosis.</p>
<h1 id="how-to-create-an-oncoplot">How to create an oncoplot?<a hidden class="anchor" aria-hidden="true" href="#how-to-create-an-oncoplot">#</a></h1>
<p>Creating an oncoplot requires few steps from data preparation to visualisation.</p>
<h2 id="data-requirements">Data requirements<a hidden class="anchor" aria-hidden="true" href="#data-requirements">#</a></h2>
<p>For this tutorial, you need genetic mutation data in MAF format. You can
refer to this previous <a href="http://localhost:1313/posts/make-maf-for-oncoplot/">post</a> on how to convert your annotated data into MAF
format with a simple piece of code, or follow vcf2maf tutorial if you have
VCF file.</p>
<p>If you want to add clinical information as metadata to the oncoplot,
you need the clinical data in tabular format:</p>
<table>
  <thead>
      <tr>
          <th style="text-align: center">Tumor_Sample_Barcode</th>
          <th style="text-align: center">Annotation1</th>
          <th style="text-align: center">Annotation2</th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: center">Same sample codes as maf</td>
          <td style="text-align: center">eg. diagnosis1</td>
          <td style="text-align: center">eg. sex</td>
      </tr>
  </tbody>
</table>
<p>Save the clinical data file with ending &ldquo;.clin&rdquo;</p>
<h2 id="tool-requirements">Tool requirements<a hidden class="anchor" aria-hidden="true" href="#tool-requirements">#</a></h2>
<p>There are several tools and packages available for creating oncoplots.
The most commonly used one is Bioconductor&rsquo;s maftools oncoplot, a popular R package.
For installation, run the following in R:</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln">1</span><span class="cl"><span class="nf">install.packages</span><span class="p">(</span><span class="s">&#34;BiocManager&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&#34;maftools&#34;</span><span class="p">)</span>
</span></span></code></pre></div><p>In python, there are some options such as package <a href="https://github.com/sbslee/fuc">fuc</a>.</p>
<h2 id="creating-oncoplot-with-maftools-oncoplot">Creating oncoplot with maftools oncoplot<a hidden class="anchor" aria-hidden="true" href="#creating-oncoplot-with-maftools-oncoplot">#</a></h2>
<h3 id="read-data">Read data<a hidden class="anchor" aria-hidden="true" href="#read-data">#</a></h3>
<p>Read in both MAF file with mutations and the optional clinical annotation file.</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln">1</span><span class="cl"><span class="n">data</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&#34;mutations.maf&#34;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="n">clindata</span> <span class="o">&lt;-</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&#34;clinicaldata.clin&#34;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#34;\t&#34;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></span></code></pre></div><p>Make any necessary filterings and changes to the file if needed. For example remember to modify VEP
annotation mutation type nomenclature to an ANNOVAR types. Check the previous post on how to make a MAF file.</p>
<h3 id="create-maftools-dataobject">Create maftools dataobject<a hidden class="anchor" aria-hidden="true" href="#create-maftools-dataobject">#</a></h3>
<p>Read files into maftools dataobject</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln">1</span><span class="cl"><span class="n">laml</span> <span class="o">&lt;-</span> <span class="nf">read.maf</span><span class="p">(</span><span class="n">maf</span> <span class="o">=</span> <span class="n">data</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl">		<span class="n">clinicalData</span> <span class="o">=</span> <span class="n">clindata</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">3</span><span class="cl">		<span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">4</span><span class="cl">
</span></span><span class="line"><span class="ln">5</span><span class="cl"><span class="nf">getClinicalData</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">laml</span><span class="p">)</span>
</span></span></code></pre></div><h3 id="set-annotation-colors">Set annotation colors<a hidden class="anchor" aria-hidden="true" href="#set-annotation-colors">#</a></h3>
<p>Maftools gives default colors for mutation types and other annotations, but if you want to change these,
you need to declare the color separately.
Also, if you have a large number of variables in some annotations, for example 15 unique diagnosis types,
it will be wise to make your own color palette.</p>
<p>If you have a reasonable number of variables in an annotation, you can use RColorBrewer and the color palettes.
Many of the RColorBrewer color palettes allow maximum of 12 colors.
To display all the available color palettes in RColorBrewer run the following lines:</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln">1</span><span class="cl"><span class="nf">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="nf">display.brewer.all</span><span class="p">()</span>
</span></span></code></pre></div><figure>
    <img loading="lazy" src="Rcolorbrewer_palettes.png"/> <figcaption>
            RColorBrewer palettes
        </figcaption>
</figure>

<p>First retrieve a number of colors from a palette you want and then assign them to the annotations.
Here&rsquo;s an example how to annotate each mutation type their own colors from the &lsquo;Paired&rsquo; palette.</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln">1</span><span class="cl"><span class="n">vc_cols</span> <span class="o">&lt;-</span> <span class="n">RColorBrewer</span><span class="o">::</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="m">6</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s">&#39;Paired&#39;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">2</span><span class="cl"><span class="nf">names</span><span class="p">(</span><span class="n">vc_cols</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span>
</span></span><span class="line"><span class="ln">3</span><span class="cl">		<span class="s">&#39;Frame_Shift_Del&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">4</span><span class="cl">		<span class="s">&#39;Missense_Mutation&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">5</span><span class="cl">		<span class="s">&#39;Nonsense_Mutation&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">6</span><span class="cl">		<span class="s">&#39;Multi_Hit&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">7</span><span class="cl">		<span class="s">&#39;Frame_shift_Ins&#39;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">8</span><span class="cl">		<span class="s">&#39;Splice_Site&#39;</span>
</span></span><span class="line"><span class="ln">9</span><span class="cl"><span class="p">)</span>
</span></span></code></pre></div><p>In case of more than 12 colors, it is probably the best to construct a palette of your own by declaring
all the colors of the palette separately. This is also a good way to construct a palette if you want
some specific or different colors for some annotations.</p>
<p>Here I constructed a list of distinct colors and assigned them to cancer diagnoses and sex.</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="n">dx_cols</span> <span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl">	<span class="s">&#34;dodgerblue2&#34;</span><span class="p">,</span> <span class="s">&#34;#E31A1C&#34;</span><span class="p">,</span> <span class="c1">#red</span>
</span></span><span class="line"><span class="ln"> 3</span><span class="cl">	<span class="s">&#34;green4&#34;</span><span class="p">,</span> <span class="s">&#34;#6A3D9A&#34;</span><span class="p">,</span> <span class="c1"># purple</span>
</span></span><span class="line"><span class="ln"> 4</span><span class="cl">	<span class="s">&#34;#FF7F00&#34;</span><span class="p">,</span> <span class="c1"># orange</span>
</span></span><span class="line"><span class="ln"> 5</span><span class="cl">  	<span class="s">&#34;aquamarine3&#34;</span><span class="p">,</span> <span class="s">&#34;gold1&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">  	<span class="s">&#34;skyblue2&#34;</span><span class="p">,</span> <span class="s">&#34;deeppink1&#34;</span><span class="p">,</span> <span class="c1"># lt pink</span>
</span></span><span class="line"><span class="ln"> 7</span><span class="cl">  	<span class="s">&#34;brown&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">  	<span class="s">&#34;#CAB2D6&#34;</span><span class="p">,</span> <span class="c1"># lt purple</span>
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">  	<span class="s">&#34;#FDBF6F&#34;</span><span class="p">,</span> <span class="c1"># lt orange</span>
</span></span><span class="line"><span class="ln">10</span><span class="cl">	<span class="s">&#34;gray70&#34;</span><span class="p">,</span> <span class="s">&#34;khaki2&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">11</span><span class="cl">  	<span class="s">&#34;maroon&#34;</span><span class="p">,</span> <span class="s">&#34;orchid1&#34;</span><span class="p">,</span> <span class="s">&#34;blue4&#34;</span><span class="p">,</span> <span class="s">&#34;steelblue4&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">12</span><span class="cl">  	<span class="s">&#34;darkturquoise&#34;</span><span class="p">,</span>  <span class="s">&#34;yellow4&#34;</span><span class="p">,</span> <span class="s">&#34;yellow3&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">13</span><span class="cl">  	<span class="s">&#34;darkorange4&#34;</span> 
</span></span><span class="line"><span class="ln">14</span><span class="cl"><span class="p">)</span>
</span></span><span class="line"><span class="ln">15</span><span class="cl">
</span></span><span class="line"><span class="ln">16</span><span class="cl"><span class="nf">names</span><span class="p">(</span><span class="n">dx_cols</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span>
</span></span><span class="line"><span class="ln">17</span><span class="cl">	<span class="s">&#34;AML&#34;</span><span class="p">,</span> <span class="s">&#34;ALL&#34;</span><span class="p">,</span> <span class="s">&#34;Breast&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">18</span><span class="cl">	<span class="s">&#34;MM&#34;</span><span class="p">,</span> <span class="s">&#34;Colon&#34;</span><span class="p">,</span> <span class="s">&#34;Lung&#34;</span><span class="p">,</span><span class="s">&#34;Skin&#34;</span><span class="p">,</span>
</span></span><span class="line"><span class="ln">19</span><span class="cl">	<span class="s">&#34;BMF&#34;</span><span class="p">,</span><span class="s">&#34;Ovary&#34;</span><span class="p">,</span><span class="s">&#34;Lymphoma&#34;</span><span class="p">,</span> 
</span></span><span class="line"><span class="ln">20</span><span class="cl">	<span class="s">&#34;Pancreas&#34;</span>
</span></span><span class="line"><span class="ln">21</span><span class="cl"><span class="p">)</span>
</span></span><span class="line"><span class="ln">22</span><span class="cl">
</span></span><span class="line"><span class="ln">23</span><span class="cl"><span class="n">sex_cols</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;skyblue2&#34;</span><span class="p">,</span> <span class="s">&#34;orchid3&#34;</span><span class="p">)</span>
</span></span><span class="line"><span class="ln">24</span><span class="cl"><span class="nf">names</span><span class="p">(</span><span class="n">sex_cols</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;M&#34;</span><span class="p">,</span><span class="s">&#34;F&#34;</span><span class="p">)</span>
</span></span></code></pre></div><p>For annotations you need to save the colors into a list before making the oncoplot.
Make sure that the list parameters (Dx_group and Sex in this case) are the same ones
you have as column names in your clinical data annotation file!</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln">1</span><span class="cl"><span class="n">anno_cols</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">Dx_group</span><span class="o">=</span><span class="n">dx_cols</span><span class="p">,</span> <span class="n">Sex</span><span class="o">=</span><span class="n">sex_cols</span>
</span></span></code></pre></div><h3 id="make-oncoplot-with-clinical-annotations">Make oncoplot with clinical annotations<a hidden class="anchor" aria-hidden="true" href="#make-oncoplot-with-clinical-annotations">#</a></h3>
<p>You have a lot of options how to modify an oncoplot and I suggest you to look into maftools oncoplot documentation
to find out what additional things you can do with oncoplots.
Here, we will draw a basic oncoplot with two different clinical annotations at the bottom of the plot.</p>
<div class="highlight my-class" id="my-codeblock" height="600"><pre tabindex="0" class="chroma"><code class="language-r" data-lang="r"><span class="line"><span class="ln"> 1</span><span class="cl"><span class="nf">oncoplot</span><span class="p">(</span><span class="n">maf</span> <span class="o">=</span> <span class="n">laml</span><span class="p">,</span> <span class="c1">#read in the maftools dataobject</span>
</span></span><span class="line"><span class="ln"> 2</span><span class="cl">	<span class="n">colors</span> <span class="o">=</span> <span class="n">vc_cols</span><span class="p">,</span> <span class="c1">#colors for mutation types</span>
</span></span><span class="line"><span class="ln"> 3</span><span class="cl">	<span class="n">clinicalFeatures</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;Dx_group&#34;</span><span class="p">,</span> <span class="s">&#34;Sex&#34;</span><span class="p">),</span> <span class="c1">#the column names of features in the annotation file</span>
</span></span><span class="line"><span class="ln"> 4</span><span class="cl">	<span class="n">annotationColor</span> <span class="o">=</span> <span class="n">anno_cols</span><span class="p">,</span> <span class="c1">#list of colors for annotations</span>
</span></span><span class="line"><span class="ln"> 5</span><span class="cl">	<span class="n">removeNonMutated</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="c1">#doesn&#39;t show samples if there are no mutations</span>
</span></span><span class="line"><span class="ln"> 6</span><span class="cl">	<span class="n">showTitle</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
</span></span><span class="line"><span class="ln"> 7</span><span class="cl">	<span class="n">top</span> <span class="o">=</span> <span class="m">10</span><span class="p">,</span> <span class="c1">#show only top 10 mutated genes</span>
</span></span><span class="line"><span class="ln"> 8</span><span class="cl">	<span class="n">showPct</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="c1">#shows percentage of individuals carrying mutation in a gene</span>
</span></span><span class="line"><span class="ln"> 9</span><span class="cl">	<span class="n">drawColBar</span> <span class="o">=</span> <span class="kc">FALSE</span> <span class="c1">#draws a barplot on top to show how many mutations each individual have</span>
</span></span><span class="line"><span class="ln">10</span><span class="cl"><span class="p">)</span>
</span></span></code></pre></div><p>And this gives a basic oncoplot with two clinical annotations. Sometimes you may need to adjust the
legend parameters in order to make all the legends with (mutation type, Dx_group and Sex should
all have their own legens underneath the oncoplot).
Parameters to adjust for this are for example: legend_height, anno_height, gene_mar, annotationFontSize.</p>
<figure>
    <img loading="lazy" src="oncoplot.jpg"/> <figcaption>
            Basic oncoplot
        </figcaption>
</figure>

<h1 id="references">References<a hidden class="anchor" aria-hidden="true" href="#references">#</a></h1>
<ul>
<li>maftools oncoplot <a href="https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/oncoplots.html">documentation</a></li>
<li>vcf2maf <a href="https://github.com/mskcc/vcf2maf">documentation</a></li>
</ul>


  </div>

  <footer class="post-footer">
    <ul class="post-tags">
      <li><a href="http://localhost:1313/tags/oncoplot/">Oncoplot</a></li>
      <li><a href="http://localhost:1313/tags/maftools/">Maftools</a></li>
    </ul>
<nav class="paginav">
  <a class="prev" href="http://localhost:1313/posts/germline-variants/">
    <span class="title">« Prev</span>
    <br>
    <span>Germline Variants vs Somatic Mutations</span>
  </a>
  <a class="next" href="http://localhost:1313/posts/make-maf-for-oncoplot/">
    <span class="title">Next »</span>
    <br>
    <span>Make Maf for Oncoplot</span>
  </a>
</nav>

  </footer>
</article>
    </main>
    
<footer class="footer">
        <span>&copy; 2024 <a href="http://localhost:1313/">Cancer Data Diaries</a></span> · 

    <span>
        Powered by
        <a href="https://gohugo.io/" rel="noopener noreferrer" target="_blank">Hugo</a> &
        <a href="https://github.com/adityatelange/hugo-PaperMod/" rel="noopener" target="_blank">PaperMod</a>
    </span>
</footer>
<a href="#top" aria-label="go to top" title="Go to Top (Alt + G)" class="top-link" id="top-link" accesskey="g">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentColor">
        <path d="M12 6H0l6-6z" />
    </svg>
</a>

<script>
    let menu = document.getElementById('menu')
    if (menu) {
        menu.scrollLeft = localStorage.getItem("menu-scroll-position");
        menu.onscroll = function () {
            localStorage.setItem("menu-scroll-position", menu.scrollLeft);
        }
    }

    document.querySelectorAll('a[href^="#"]').forEach(anchor => {
        anchor.addEventListener("click", function (e) {
            e.preventDefault();
            var id = this.getAttribute("href").substr(1);
            if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({
                    behavior: "smooth"
                });
            } else {
                document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();
            }
            if (id === "top") {
                history.replaceState(null, null, " ");
            } else {
                history.pushState(null, null, `#${id}`);
            }
        });
    });

</script>
<script>
    var mybutton = document.getElementById("top-link");
    window.onscroll = function () {
        if (document.body.scrollTop > 800 || document.documentElement.scrollTop > 800) {
            mybutton.style.visibility = "visible";
            mybutton.style.opacity = "1";
        } else {
            mybutton.style.visibility = "hidden";
            mybutton.style.opacity = "0";
        }
    };

</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

</script>
<script>
    document.querySelectorAll('pre > code').forEach((codeblock) => {
        const container = codeblock.parentNode.parentNode;

        const copybutton = document.createElement('button');
        copybutton.classList.add('copy-code');
        copybutton.innerHTML = 'copy';

        function copyingDone() {
            copybutton.innerHTML = 'copied!';
            setTimeout(() => {
                copybutton.innerHTML = 'copy';
            }, 2000);
        }

        copybutton.addEventListener('click', (cb) => {
            if ('clipboard' in navigator) {
                navigator.clipboard.writeText(codeblock.textContent);
                copyingDone();
                return;
            }

            const range = document.createRange();
            range.selectNodeContents(codeblock);
            const selection = window.getSelection();
            selection.removeAllRanges();
            selection.addRange(range);
            try {
                document.execCommand('copy');
                copyingDone();
            } catch (e) { };
            selection.removeRange(range);
        });

        if (container.classList.contains("highlight")) {
            container.appendChild(copybutton);
        } else if (container.parentNode.firstChild == container) {
            
        } else if (codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName == "TABLE") {
            
            codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(copybutton);
        } else {
            
            codeblock.parentNode.appendChild(copybutton);
        }
    });
</script>
</body>

</html>
